---
permalink: /
title: "About me"
excerpt: "About me"
author_profile: true
redirect_from: 
  - /about/
  - /about.html
---

Hi, I'm Dongyuan, a new tenure track assistant professor in [Department of Genetics and Genome Sciences, University of Connecticut, Health Center (UConn Health)](https://health.uconn.edu/genetics/).

Previously, I got my Ph.D. in [Bioinformatics](https://bioinformatics.ucla.edu/), at University of California, Los Angeles (UCLA), under the supervision of [Dr. Jingyi Jessica Li](http://jsb.ucla.edu/) from [Department of Statistics & Data Science](http://statistics.ucla.edu/). Previously, I received Master of Science in [Computational Biology & Quantitative Genetics](https://www.hsph.harvard.edu/sm-computational-biology/) from Department of Biostatistcs, Harvard T.H. Chan School of Public Health, under supervision of [Dr. Rafael Irizarry](https://rafalab.dfci.harvard.edu/). I obtained Bachelor of Science in [Biological Science](https://life.fudan.edu.cn/) from Fudan University, Shanghai, China, under supervision of [Dr. Bao-Rong Lu](https://life.fudan.edu.cn/67/92/c31280a354194/page.htm).

I am looking for students and postdocs to work on computational genomics. Please email <dosong@uchc.edu> if you are interested in. Perspective student should first apply [Ph.D. in Biomedical Science](https://health.uconn.edu/graduate-school/academics/programs/ph-d-biomedical-science/). The deadline of this year is **12/01/2024**. Current students in Ph.D. in Biomedical Science are welcome for rotation. For more details, please check our group website [DS lab](https://dsong-lab.github.io/).

My research focuses on developing computational tools in analyzing single-cell and spatial omics. Some of my previous works include:
1. Probabilistic generative models of high-dimensional single-cell and spatial multi-omics data. I develped [scDesign3 (Nature Biotechnology, 2024)](https://doi.org/10.1038/s41587-023-01772-1), an "all-in-one" multimodal single-cell and spatial omics simulator which summarizes the input real dataset into a parametric model. I also contributed to the development of [scReadSim (Nature Communications, 2023)](https://www.biorxiv.org/content/10.1101/2022.05.29.493924v3), a single-cell RNA-seq and ATAC-seq read simulator, and [scDesign2 (Genome Biology, 2021)](https://doi.org/10.1186/s13059-021-02367-2), the predecessor of scDesign3.
2. Differential expression (DE) test and false discovery rate (FDR). We developed [ClusterDE (bioRxiv, 2023)](https://www.biorxiv.org/content/10.1101/2023.07.21.550107v1), a post-clustering DE method controlling FDR under "double dipping (i.e., first clustering then DE between clusters)". Previously, I developed [PseudotimeDE (Genome Biology, 2021)](https://doi.org/10.1186/s13059-021-02341-y), a DE method for testing gene changes along cell pseudotime accounting for the uncertainty of pseudotime.
3. Informative genes/cells selection in large-scale scRNA-seq data. We developed [scPNMF (Bioinformatics, 2021)](https://doi.org/10.1093/bioinformatics/btab273), an informative gene selection method for selecting only a small number of genes. We also developed [scSampler (Bioinformatics, 2022)](https://doi.org/10.1093/bioinformatics/btac271), a diveristy-preserving cell subsampling method for large-scale datasets.
