---
permalink: /
title: "About me"
excerpt: "About me"
author_profile: true
redirect_from: 
  - /about/
  - /about.html
---

Hi, I'm Dongyuan, Ph.D. in [Bioinformatics](https://bioinformatics.ucla.edu/), at University of California, Los Angeles (UCLA). I am under supervision of [Dr. Jingyi Jessica Li](http://jsb.ucla.edu/) from [Department of Statistics & Data Science](http://statistics.ucla.edu/). Previously, I received Master of Science in [Computational Biology & Quantitative Genetics](https://www.hsph.harvard.edu/sm-computational-biology/) from Department of Biostatistcs, Harvard T.H. Chan School of Public Health, under supervision of [Dr. Rafael Irizarry](https://rafalab.dfci.harvard.edu/). I obtained Bachelor of Science in [Biological Science](https://life.fudan.edu.cn/) from Fudan University, Shanghai, China, under supervision of [Dr. Bao-Rong Lu](https://life.fudan.edu.cn/67/92/c31280a354194/page.htm).

**On November 1st, 2024, I will begin my appointment as a tenure track assistant professor in Department of Genetics and Genome Sciences, University of Connecticut School of Medicine.**

My research focuses on developing computational tools in analyzing high-dimensional single-cell and spatial omics. Some of my major works include:
1. Probabilistic generative models of high-dimensional single-cell and spatial multi-omics data. I develped [scDesign3](https://doi.org/10.1038/s41587-023-01772-1), an "all-in-one" multimodal single-cell and spatial omics simulator which summarizes the input real dataset into a parametric model. I also contributed to the development of [scReadSim](https://www.biorxiv.org/content/10.1101/2022.05.29.493924v3), a single-cell RNA-seq and ATAC-seq read simulator, and [scDesign2](https://doi.org/10.1186/s13059-021-02367-2), the predecessor of scDesign3.
2. Differential expression (DE) test and false discovery rate (FDR) control based on inferred variables. We developed [ClusterDE](https://www.biorxiv.org/content/10.1101/2023.07.21.550107v1), a post-clustering DE method controlling FDR under "double dipping (i.e., first clustering then DE between clusters)". Previously, I developed [PseudotimeDE](https://doi.org/10.1186/s13059-021-02341-y), a DE method for testing gene changes along cell pseudotime accounting for the uncertainty of pseudotime.
3. Informative genes/cells selection in large-scale scRNA-seq data. We developed [scPNMF](https://doi.org/10.1093/bioinformatics/btab273), an informative gene selection method for selecting only a small number of genes. We also developed [scSampler](https://doi.org/10.1093/bioinformatics/btac271), a diveristy-preserving cell subsampling method for large datasets.
