<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="scDesign3"><title>Benchmarking DE Analysis with scDesign3 • scDesign3</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Benchmarking DE Analysis with scDesign3"><meta property="og:description" content="scDesign3"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-article">

<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Benchmarking DE Analysis with scDesign3</h1>
                        <h4 data-toc-skip class="author">Dongyuan Song</h4>
            <address class="author_afil">
      Bioinformatics IDP, University of California, Los Angeles<br>            <a class="author_email" href="mailto:#"><a href="mailto:dongyuansong@ucla.edu" class="email">dongyuansong@ucla.edu</a></a>
      </address>
                              <h4 data-toc-skip class="author">Qingyang Wang</h4>
            <address class="author_afil">
      Department of Statistics, University of California, Los Angeles<br>            <a class="author_email" href="mailto:#"><a href="mailto:qw802@g.ucla.edu" class="email">qw802@g.ucla.edu</a></a>
      </address>
                  
            <h4 data-toc-skip class="date">17 January 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/../../scDesign3/code/vignettes/scDesign3-DEanalysis-vignette.Rmd" class="external-link"><code>../../scDesign3/code/vignettes/scDesign3-DEanalysis-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>scDesign3-DEanalysis-vignette.Rmd</code></div>
    </div>

    
    
<style type="text/css">
pre {
  white-space: pre !important;
  overflow-x: scroll !important;
}
</style>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">library</span>(scDesign3)</a>
<a class="sourceLine" id="cb1-2" title="2"><span class="kw">library</span>(SingleCellExperiment)</a>
<a class="sourceLine" id="cb1-3" title="3"><span class="kw">library</span>(ggplot2)</a>
<a class="sourceLine" id="cb1-4" title="4"><span class="kw">library</span>(DuoClustering2018)</a>
<a class="sourceLine" id="cb1-5" title="5"><span class="kw">library</span>(Seurat)</a>
<a class="sourceLine" id="cb1-6" title="6"><span class="kw">library</span>(SeuratObject)</a>
<a class="sourceLine" id="cb1-7" title="7"><span class="kw">library</span>(scran)</a>
<a class="sourceLine" id="cb1-8" title="8"><span class="kw">library</span>(parallel)</a>
<a class="sourceLine" id="cb1-9" title="9"><span class="kw">library</span>(monocle3)</a>
<a class="sourceLine" id="cb1-10" title="10"><span class="kw">library</span>(DESeq2)</a>
<a class="sourceLine" id="cb1-11" title="11"><span class="kw">library</span>(BiocParallel)</a>
<a class="sourceLine" id="cb1-12" title="12"><span class="kw">library</span>(NBAMSeq)</a>
<a class="sourceLine" id="cb1-13" title="13"><span class="kw">library</span>(PseudotimeDE)</a>
<a class="sourceLine" id="cb1-14" title="14"><span class="kw">library</span>(dplyr)</a>
<a class="sourceLine" id="cb1-15" title="15"><span class="kw">library</span>(tradeSeq)</a>
<a class="sourceLine" id="cb1-16" title="16"><span class="kw">library</span>(reshape2)</a>
<a class="sourceLine" id="cb1-17" title="17"><span class="kw">library</span>(spatialDE)</a>
<a class="sourceLine" id="cb1-18" title="18"><span class="kw">library</span>(SPARK)</a>
<a class="sourceLine" id="cb1-19" title="19"><span class="kw">library</span>(scales)</a>
<a class="sourceLine" id="cb1-20" title="20"><span class="kw">library</span>(tidyr)</a>
<a class="sourceLine" id="cb1-21" title="21"><span class="kw">theme_set</span>(<span class="kw">theme_bw</span>())</a></code></pre></div>
<p>In this tutorial, we will demonstrate how to use scDesign3 to generate negative control and benchmark methods for identifying differentially expressed (DE) genes between discrete cell types or continuous trajectory from scRNA-seq data, and identifying spatially variable genes (SVG) from spatial transcriptomics data. Please note that here we only did a very brief benchmarking on a few methods for illustration purpose, not for formal comparison.</p>
<div id="identification-of-de-genes-between-discrete-cell-types" class="section level2">
<h2>Identification of DE genes between discrete cell types</h2>
<div id="read-in-the-reference-data" class="section level3">
<h3>Read in the reference data</h3>
<p>The raw data is from the R package <code>DuoClustering2018</code> which contain a set of datasets with various clustering results.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1">Zhengmix4eq_sce &lt;-<span class="st"> </span><span class="kw">get</span>(<span class="st">&quot;sce_filteredExpr10_Zhengmix4eq&quot;</span>)(<span class="dt">metadata =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>The top 200 highly variable genes are kept for generating synthetic data.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1">ngene &lt;-<span class="st"> </span><span class="dv">200</span></a>
<a class="sourceLine" id="cb3-2" title="2"><span class="kw">logcounts</span>(Zhengmix4eq_sce) &lt;-<span class="st"> </span><span class="kw">log1p</span>(<span class="kw">counts</span>(Zhengmix4eq_sce))</a>
<a class="sourceLine" id="cb3-3" title="3">dec.seurat_obj &lt;-<span class="st"> </span><span class="kw">modelGeneVar</span>(Zhengmix4eq_sce)</a>
<a class="sourceLine" id="cb3-4" title="4">chosen &lt;-<span class="st"> </span><span class="kw">getTopHVGs</span>(Zhengmix4eq_sce, <span class="dt">n =</span> ngene)</a>
<a class="sourceLine" id="cb3-5" title="5">example_sce &lt;-<span class="st"> </span>Zhengmix4eq_sce[chosen,]</a></code></pre></div>
<p>We extract out B cells and regulatory T cells only and use all cells from these two cell types to simulate synthetic data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1">selected_cells &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">colData</span>(example_sce)<span class="op">$</span>phenoid <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;b.cells&quot;</span>,<span class="st">&quot;regulatory.t&quot;</span>))</a>
<a class="sourceLine" id="cb4-2" title="2">example_sce &lt;-<span class="st"> </span>example_sce[,selected_cells]</a>
<a class="sourceLine" id="cb4-3" title="3"><span class="kw">colData</span>(example_sce)<span class="op">$</span>cell_type &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">colData</span>(example_sce)<span class="op">$</span>phenoid)</a>
<a class="sourceLine" id="cb4-4" title="4">example_sce</a>
<a class="sourceLine" id="cb4-5" title="5"><span class="kw">head</span>(<span class="kw">colData</span>(example_sce))</a></code></pre></div>
</div>
<div id="simulation" class="section level3">
<h3>Simulation</h3>
<p>We use the step-by-step functions instead of the one-shot function to generate synthetic data since these step-by-step functions allow us to alter estimated parameters and generate new data based on our desired parameters.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1">example_data &lt;-<span class="st"> </span><span class="kw">construct_data</span>(</a>
<a class="sourceLine" id="cb5-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb5-3" title="3">    <span class="dt">assay_use =</span> <span class="st">&quot;counts&quot;</span>,</a>
<a class="sourceLine" id="cb5-4" title="4">    <span class="dt">celltype =</span> <span class="st">&quot;cell_type&quot;</span>,</a>
<a class="sourceLine" id="cb5-5" title="5">    <span class="dt">pseudotime =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb5-6" title="6">    <span class="dt">spatial =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb5-7" title="7">    <span class="dt">other_covariates =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb5-8" title="8">    <span class="dt">corr_by =</span> <span class="st">&quot;1&quot;</span></a>
<a class="sourceLine" id="cb5-9" title="9">  )</a></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1">example_marginal &lt;-<span class="st"> </span><span class="kw">fit_marginal</span>(</a>
<a class="sourceLine" id="cb6-2" title="2">    <span class="dt">data =</span> example_data,</a>
<a class="sourceLine" id="cb6-3" title="3">    <span class="dt">predictor =</span> <span class="st">&quot;gene&quot;</span>,</a>
<a class="sourceLine" id="cb6-4" title="4">    <span class="dt">mu_formula =</span> <span class="st">&quot;cell_type&quot;</span>,</a>
<a class="sourceLine" id="cb6-5" title="5">    <span class="dt">sigma_formula =</span> <span class="st">&quot;1&quot;</span>,</a>
<a class="sourceLine" id="cb6-6" title="6">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb6-7" title="7">    <span class="dt">n_cores =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb6-8" title="8">    <span class="dt">usebam =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb6-9" title="9">  )</a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">example_copula &lt;-<span class="st"> </span><span class="kw">fit_copula</span>(</a>
<a class="sourceLine" id="cb7-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb7-3" title="3">    <span class="dt">assay_use =</span> <span class="st">&quot;counts&quot;</span>,</a>
<a class="sourceLine" id="cb7-4" title="4">    <span class="dt">marginal_list =</span> example_marginal,</a>
<a class="sourceLine" id="cb7-5" title="5">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb7-6" title="6">    <span class="dt">copula =</span> <span class="st">&quot;gaussian&quot;</span>,</a>
<a class="sourceLine" id="cb7-7" title="7">    <span class="dt">n_cores =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb7-8" title="8">    <span class="dt">new_covariate =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb7-9" title="9">    <span class="dt">input_data =</span> example_data<span class="op">$</span>dat</a>
<a class="sourceLine" id="cb7-10" title="10">  )</a></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1">example_para &lt;-<span class="st"> </span><span class="kw">extract_para</span>(</a>
<a class="sourceLine" id="cb8-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb8-3" title="3">    <span class="dt">marginal_list =</span> example_marginal,</a>
<a class="sourceLine" id="cb8-4" title="4">    <span class="dt">n_cores =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb8-5" title="5">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb8-6" title="6">    <span class="dt">new_covariate =</span> <span class="ot">NULL</span></a>
<a class="sourceLine" id="cb8-7" title="7">  )</a></code></pre></div>
<p>Here, we examine the <code>mean_mat</code>, which is one of the outputs from the previous function <code>extract_para()</code>. For each gene, we calculate the difference in the between the maximum mean parameter and minimum mean parameter across all cells. We select genes which the gene’s mean difference across cells are in the top 50 largest differences. We regard these genes as DE genes. Then, we manually set the mean parameters of the rest genes to be the same across all cells. We regard all genes with the same mean parameter across cells as non-DE genes. Of course, this is a very flexible step and users may choose other ideas to modify the mean matrix.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">diff &lt;-<span class="st"> </span><span class="kw">apply</span>(example_para<span class="op">$</span>mean_mat, <span class="dv">2</span>, <span class="cf">function</span>(x){<span class="kw">max</span>(x)<span class="op">-</span><span class="kw">min</span>(x)})</a>
<a class="sourceLine" id="cb9-2" title="2">diff_ordered &lt;-<span class="st"> </span><span class="kw">order</span>(diff, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-3" title="3">diff &lt;-<span class="st"> </span>diff[diff_ordered]</a>
<a class="sourceLine" id="cb9-4" title="4">num_de &lt;-<span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb9-5" title="5">de_idx &lt;-<span class="st"> </span><span class="kw">names</span>(diff[<span class="dv">1</span><span class="op">:</span>num_de])</a>
<a class="sourceLine" id="cb9-6" title="6">non_de_idx &lt;-<span class="st"> </span><span class="kw">names</span>(diff[<span class="op">-</span>(<span class="dv">1</span><span class="op">:</span>num_de)])</a>
<a class="sourceLine" id="cb9-7" title="7">non_de_mat &lt;-<span class="st"> </span><span class="kw">apply</span>(example_para<span class="op">$</span>mean_mat[,non_de_idx], <span class="dv">2</span>, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb9-8" title="8">  avg &lt;-<span class="st"> </span>(<span class="kw">max</span>(x)<span class="op">+</span><span class="kw">min</span>(x))<span class="op">/</span><span class="dv">2</span></a>
<a class="sourceLine" id="cb9-9" title="9">  new_mean &lt;-<span class="st"> </span><span class="kw">rep</span>(avg, <span class="kw">length</span>(x))</a>
<a class="sourceLine" id="cb9-10" title="10">  <span class="kw">return</span>(new_mean)</a>
<a class="sourceLine" id="cb9-11" title="11">})</a>
<a class="sourceLine" id="cb9-12" title="12">example_para<span class="op">$</span>mean_mat[,non_de_idx] &lt;-<span class="st"> </span>non_de_mat</a></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1"> example_newcount &lt;-<span class="st"> </span><span class="kw">simu_new</span>(</a>
<a class="sourceLine" id="cb10-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb10-3" title="3">    <span class="dt">mean_mat =</span> example_para<span class="op">$</span>mean_mat,</a>
<a class="sourceLine" id="cb10-4" title="4">    <span class="dt">sigma_mat =</span> example_para<span class="op">$</span>sigma_mat,</a>
<a class="sourceLine" id="cb10-5" title="5">    <span class="dt">zero_mat =</span> example_para<span class="op">$</span>zero_mat,</a>
<a class="sourceLine" id="cb10-6" title="6">    <span class="dt">quantile_mat =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb10-7" title="7">    <span class="dt">copula_list =</span> example_copula<span class="op">$</span>copula_list,</a>
<a class="sourceLine" id="cb10-8" title="8">    <span class="dt">n_cores =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb10-9" title="9">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb10-10" title="10">    <span class="dt">input_data =</span> example_data<span class="op">$</span>dat,</a>
<a class="sourceLine" id="cb10-11" title="11">    <span class="dt">new_covariate =</span> example_data<span class="op">$</span>new_covariate,</a>
<a class="sourceLine" id="cb10-12" title="12">    <span class="dt">important_feature =</span> example_copula<span class="op">$</span>important_feature</a>
<a class="sourceLine" id="cb10-13" title="13">  )</a></code></pre></div>
</div>
<div id="de-test" class="section level3">
<h3>DE Test</h3>
<p>Then, we follow <a href="https://satijalab.org/seurat/articles/seurat_obj3k_tutorial.html">Seurat’s pipeline</a> to preprocess the simulated data.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">seurat_obj &lt;-<span class="st"> </span><span class="kw">CreateSeuratObject</span>(<span class="dt">counts =</span> example_newcount, <span class="dt">project =</span> <span class="st">&quot;seurat_obj&quot;</span>, <span class="dt">min.cells =</span> <span class="dv">0</span>, <span class="dt">min.features =</span> <span class="dv">0</span>)</a>
<a class="sourceLine" id="cb11-2" title="2">seurat_obj[[<span class="st">&quot;percent.mt&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">PercentageFeatureSet</span>(seurat_obj, <span class="dt">pattern =</span> <span class="st">&quot;^MT-&quot;</span>)</a>
<a class="sourceLine" id="cb11-3" title="3">seurat_obj &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(seurat_obj, <span class="dt">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="dt">scale.factor =</span> <span class="dv">10000</span>)</a>
<a class="sourceLine" id="cb11-4" title="4">seurat_obj &lt;-<span class="st"> </span><span class="kw">FindVariableFeatures</span>(seurat_obj, <span class="dt">selection.method =</span> <span class="st">&quot;vst&quot;</span>,<span class="dt">nfeatures =</span> <span class="dv">200</span>)</a>
<a class="sourceLine" id="cb11-5" title="5">all.genes &lt;-<span class="st"> </span><span class="kw">rownames</span>(seurat_obj)</a>
<a class="sourceLine" id="cb11-6" title="6">seurat_obj &lt;-<span class="st"> </span><span class="kw">ScaleData</span>(seurat_obj, <span class="dt">features =</span> all.genes)</a>
<a class="sourceLine" id="cb11-7" title="7">seurat_obj &lt;-<span class="st"> </span><span class="kw">RunPCA</span>(seurat_obj, <span class="dt">features =</span> <span class="kw">VariableFeatures</span>(<span class="dt">object =</span> seurat_obj))</a>
<a class="sourceLine" id="cb11-8" title="8">seurat_obj &lt;-<span class="st"> </span><span class="kw">JackStraw</span>(seurat_obj, <span class="dt">num.replicate =</span> <span class="dv">100</span>)</a>
<a class="sourceLine" id="cb11-9" title="9">seurat_obj &lt;-<span class="st"> </span><span class="kw">ScoreJackStraw</span>(seurat_obj, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>)</a></code></pre></div>
<p>Since we already have the ground truth cell type annotations for our simulated dataset, we can directly use the cell type annotations we have instead of running <code>FindClusters</code> from <code>Seurat</code> to avoid the double-dipping issue.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1">seurat_obj &lt;-<span class="st"> </span><span class="kw">FindNeighbors</span>(seurat_obj, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>)</a>
<a class="sourceLine" id="cb12-2" title="2">ct &lt;-<span class="st"> </span><span class="kw">colData</span>(example_sce)<span class="op">$</span>cell_type</a>
<a class="sourceLine" id="cb12-3" title="3"><span class="kw">names</span>(ct) &lt;-<span class="st"> </span><span class="kw">colnames</span>(example_sce)</a>
<a class="sourceLine" id="cb12-4" title="4">seurat_obj[[<span class="st">&quot;cell_type&quot;</span>]] &lt;-<span class="st"> </span>ct</a>
<a class="sourceLine" id="cb12-5" title="5"><span class="kw">Idents</span>(seurat_obj) &lt;-<span class="st"> &quot;cell_type&quot;</span></a>
<a class="sourceLine" id="cb12-6" title="6">seurat_obj &lt;-<span class="st"> </span><span class="kw">RunUMAP</span>(seurat_obj, <span class="dt">dims =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>)</a></code></pre></div>
<p>Then, we follow <a href="https://satijalab.org/seurat/articles/de_vignette.html">Seurat’s tutorial</a> to conduct DE test.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" title="1">test &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;wilcox&quot;</span>, <span class="st">&quot;bimod&quot;</span>, <span class="st">&quot;t&quot;</span>, <span class="st">&quot;poisson&quot;</span>, <span class="st">&quot;negbinom&quot;</span>, <span class="st">&quot;LR&quot;</span>, <span class="st">&quot;MAST&quot;</span>, <span class="st">&quot;DESeq2&quot;</span>)</a>
<a class="sourceLine" id="cb13-2" title="2">qvals &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="kw">dim</span>(seurat_obj)[<span class="dv">1</span>], <span class="dt">ncol =</span> <span class="kw">length</span>(test))</a>
<a class="sourceLine" id="cb13-3" title="3"><span class="cf">for</span> (x <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(test)) {</a>
<a class="sourceLine" id="cb13-4" title="4">  markers &lt;-<span class="st"> </span><span class="kw">FindMarkers</span>(seurat_obj, <span class="dt">ident.1 =</span> <span class="st">&quot;b.cells&quot;</span>, <span class="dt">ident.2 =</span> <span class="st">&quot;regulatory.t&quot;</span>, <span class="dt">test.use =</span> test[x],</a>
<a class="sourceLine" id="cb13-5" title="5">                         <span class="dt">logfc.threshold =</span> <span class="dv">0</span>, <span class="dt">min.pct =</span> <span class="dv">0</span>, <span class="dt">min.cells.feature =</span> <span class="dv">1</span>, <span class="dt">min.cells.group =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb13-6" title="6">  qvals[,x] &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(markers[<span class="kw">rownames</span>(seurat_obj),<span class="st">&quot;p_val&quot;</span>], <span class="dt">method =</span> <span class="st">&quot;BH&quot;</span>, <span class="kw">length</span>(<span class="kw">rownames</span>(seurat_obj)))</a>
<a class="sourceLine" id="cb13-7" title="7">}</a>
<a class="sourceLine" id="cb13-8" title="8"><span class="kw">colnames</span>(qvals) &lt;-<span class="st"> </span>test</a>
<a class="sourceLine" id="cb13-9" title="9"><span class="kw">rownames</span>(qvals) &lt;-<span class="st"> </span><span class="kw">rownames</span>(seurat_obj)</a></code></pre></div>
<p>Since we manually created non-DE genes in the <code>extra_para()</code> step, now we can calculate the actual false discovery proportion(FDP) and power of the DE tests we conducted above with various target FDR threshold.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1">targetFDR &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">seq</span>(<span class="fl">0.01</span>,<span class="fl">0.1</span>,<span class="dt">by=</span><span class="fl">0.01</span>),<span class="kw">seq</span>(<span class="fl">0.2</span>,<span class="fl">0.5</span>,<span class="dt">by=</span><span class="fl">0.1</span>))</a>
<a class="sourceLine" id="cb14-2" title="2">de &lt;-de_idx</a>
<a class="sourceLine" id="cb14-3" title="3">fdp_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(targetFDR), <span class="dt">ncol =</span> <span class="kw">length</span>(test))</a>
<a class="sourceLine" id="cb14-4" title="4"><span class="kw">colnames</span>(fdp_mat) &lt;-<span class="st"> </span>test</a>
<a class="sourceLine" id="cb14-5" title="5"><span class="kw">rownames</span>(fdp_mat) &lt;-<span class="st"> </span>targetFDR</a>
<a class="sourceLine" id="cb14-6" title="6">power_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(targetFDR), <span class="dt">ncol =</span> <span class="kw">length</span>(test))</a>
<a class="sourceLine" id="cb14-7" title="7"><span class="kw">colnames</span>(power_mat) &lt;-<span class="st"> </span>test</a>
<a class="sourceLine" id="cb14-8" title="8"><span class="kw">rownames</span>(power_mat) &lt;-<span class="st"> </span>targetFDR</a>
<a class="sourceLine" id="cb14-9" title="9"></a>
<a class="sourceLine" id="cb14-10" title="10"><span class="cf">for</span> (t <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(test)) {</a>
<a class="sourceLine" id="cb14-11" title="11">  curr_p =<span class="st"> </span>qvals[,t]</a>
<a class="sourceLine" id="cb14-12" title="12">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(targetFDR)) {</a>
<a class="sourceLine" id="cb14-13" title="13">    thre &lt;-<span class="st"> </span>targetFDR[i]</a>
<a class="sourceLine" id="cb14-14" title="14">    discovery &lt;-<span class="st"> </span><span class="kw">which</span>(curr_p <span class="op">&lt;=</span><span class="st"> </span>thre)</a>
<a class="sourceLine" id="cb14-15" title="15">    tp &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(<span class="kw">names</span>(discovery),de))</a>
<a class="sourceLine" id="cb14-16" title="16">    <span class="cf">if</span>(<span class="kw">length</span>(discovery) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>){</a>
<a class="sourceLine" id="cb14-17" title="17">      fdp &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb14-18" title="18">    }<span class="cf">else</span>{</a>
<a class="sourceLine" id="cb14-19" title="19">      fdp &lt;-<span class="st"> </span>(<span class="kw">length</span>(discovery) <span class="op">-</span><span class="st"> </span>tp)<span class="op">/</span><span class="kw">length</span>(discovery)</a>
<a class="sourceLine" id="cb14-20" title="20">    }</a>
<a class="sourceLine" id="cb14-21" title="21">    power &lt;-<span class="st"> </span>tp<span class="op">/</span><span class="kw">length</span>(de)</a>
<a class="sourceLine" id="cb14-22" title="22">    fdp_mat[i, t] &lt;-<span class="st"> </span>fdp</a>
<a class="sourceLine" id="cb14-23" title="23">    power_mat[i,t] &lt;-<span class="st"> </span>power</a>
<a class="sourceLine" id="cb14-24" title="24">  }</a>
<a class="sourceLine" id="cb14-25" title="25">}</a></code></pre></div>
<p>Lastly, we visualize the Target FDR vs Actual FDP and Target FDR vs Power below.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" title="1">fdp_long &lt;-<span class="st"> </span><span class="kw">melt</span>(fdp_mat)</a>
<a class="sourceLine" id="cb15-2" title="2"><span class="kw">colnames</span>(fdp_long) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Target FDR&quot;</span>,<span class="st">&quot;test_method&quot;</span>,<span class="st">&quot;Actual FDP&quot;</span>)</a>
<a class="sourceLine" id="cb15-3" title="3">fdp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(fdp_long) <span class="op">+</span></a>
<a class="sourceLine" id="cb15-4" title="4"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span><span class="st">`</span><span class="dt">Actual FDP</span><span class="st">`</span>,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb15-5" title="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span><span class="st">`</span><span class="dt">Actual FDP</span><span class="st">`</span>,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb15-6" title="6"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>,<span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>,<span class="dt">color=</span><span class="st">&quot;grey&quot;</span>)<span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb15-7" title="7"><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">expand_limits</span>(<span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</a>
<a class="sourceLine" id="cb15-8" title="8">fdp_plot</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-17-1.png" width="700" /></p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1">power_long &lt;-<span class="st"> </span><span class="kw">melt</span>(power_mat)</a>
<a class="sourceLine" id="cb16-2" title="2"><span class="kw">colnames</span>(power_long) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Target FDR&quot;</span>,<span class="st">&quot;test_method&quot;</span>,<span class="st">&quot;Power&quot;</span>)</a>
<a class="sourceLine" id="cb16-3" title="3">power_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(power_long) <span class="op">+</span></a>
<a class="sourceLine" id="cb16-4" title="4"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span>Power,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb16-5" title="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span>Power,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb16-6" title="6"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb16-7" title="7">power_plot</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-18-1.png" width="700" /></p>
</div>
</div>
<div id="identication-of-de-genes-along-a-trajectory" class="section level2">
<h2>Identication of DE genes along a trajectory</h2>
<div id="read-in-the-reference-data-1" class="section level3">
<h3>Read in the reference data</h3>
<p>The raw data is from the <a href="https://scvelo.readthedocs.io/scvelo.datasets.pancreas/">scvelo</a>, which describes pancreatic endocrinogenesis. We pre-select the top 1000 highly variable genes and filter out some cell types to ensure a <strong>single trajectory</strong>.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" title="1">example_sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>((<span class="kw">url</span>(<span class="st">&quot;https://www.dropbox.com/s/p1mvl5osxp55sot/PANCREAS_sce.rds?raw=1&quot;</span>)))</a>
<a class="sourceLine" id="cb17-2" title="2"><span class="kw">print</span>(example_sce)</a>
<a class="sourceLine" id="cb17-3" title="3"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb17-4" title="4"><span class="co">#&gt; dim: 1000 2087 </span></a>
<a class="sourceLine" id="cb17-5" title="5"><span class="co">#&gt; metadata(5): clusters_coarse_colors clusters_colors day_colors</span></a>
<a class="sourceLine" id="cb17-6" title="6"><span class="co">#&gt;   neighbors pca</span></a>
<a class="sourceLine" id="cb17-7" title="7"><span class="co">#&gt; assays(6): X spliced ... cpm logcounts</span></a>
<a class="sourceLine" id="cb17-8" title="8"><span class="co">#&gt; rownames(1000): Pyy Iapp ... Eya2 Kif21a</span></a>
<a class="sourceLine" id="cb17-9" title="9"><span class="co">#&gt; rowData names(1): highly_variable_genes</span></a>
<a class="sourceLine" id="cb17-10" title="10"><span class="co">#&gt; colnames(2087): AAACCTGAGAGGGATA AAACCTGGTAAGTGGC ... TTTGTCAAGTGACATA</span></a>
<a class="sourceLine" id="cb17-11" title="11"><span class="co">#&gt;   TTTGTCAAGTGTGGCA</span></a>
<a class="sourceLine" id="cb17-12" title="12"><span class="co">#&gt; colData names(7): clusters_coarse clusters ... sizeFactor pseudotime</span></a>
<a class="sourceLine" id="cb17-13" title="13"><span class="co">#&gt; reducedDimNames(4): X_pca X_umap PCA UMAP</span></a>
<a class="sourceLine" id="cb17-14" title="14"><span class="co">#&gt; mainExpName: NULL</span></a>
<a class="sourceLine" id="cb17-15" title="15"><span class="co">#&gt; altExpNames(0):</span></a></code></pre></div>
<p>To save computational time, we only use the top 200 genes.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1">example_sce &lt;-<span class="st"> </span>example_sce[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>, ]</a></code></pre></div>
</div>
<div id="simulation-1" class="section level3">
<h3>Simulation</h3>
<p>We use the step-by-step functions instead of the one-shot function to generate synthetic data since these step-by-step functions allow us to alter estimated parameters and generate new data based on our desired parameters.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1">PANCREAS_data &lt;-<span class="st"> </span><span class="kw">construct_data</span>(</a>
<a class="sourceLine" id="cb19-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb19-3" title="3">    <span class="dt">assay_use =</span> <span class="st">&quot;counts&quot;</span>,</a>
<a class="sourceLine" id="cb19-4" title="4">    <span class="dt">celltype =</span> <span class="st">&quot;cell_type&quot;</span>,</a>
<a class="sourceLine" id="cb19-5" title="5">    <span class="dt">pseudotime =</span> <span class="st">&quot;pseudotime&quot;</span>,</a>
<a class="sourceLine" id="cb19-6" title="6">    <span class="dt">spatial =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb19-7" title="7">    <span class="dt">other_covariates =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb19-8" title="8">    <span class="dt">corr_by =</span> <span class="st">&quot;1&quot;</span></a>
<a class="sourceLine" id="cb19-9" title="9">  )</a></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb20-1" title="1">PANCREAS_marginal &lt;-<span class="st"> </span><span class="kw">fit_marginal</span>(</a>
<a class="sourceLine" id="cb20-2" title="2">    <span class="dt">data =</span> PANCREAS_data,</a>
<a class="sourceLine" id="cb20-3" title="3">    <span class="dt">predictor =</span> <span class="st">&quot;gene&quot;</span>,</a>
<a class="sourceLine" id="cb20-4" title="4">    <span class="dt">mu_formula =</span> <span class="st">&quot;s(pseudotime, k = 10, bs = &#39;cr&#39;)&quot;</span>,</a>
<a class="sourceLine" id="cb20-5" title="5">    <span class="dt">sigma_formula =</span> <span class="st">&quot;s(pseudotime, k = 5, bs = &#39;cr&#39;)&quot;</span>,</a>
<a class="sourceLine" id="cb20-6" title="6">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb20-7" title="7">    <span class="dt">n_cores =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb20-8" title="8">    <span class="dt">usebam =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb20-9" title="9">  )</a></code></pre></div>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" title="1">PANCREAS_copula &lt;-<span class="st"> </span><span class="kw">fit_copula</span>(</a>
<a class="sourceLine" id="cb21-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb21-3" title="3">    <span class="dt">assay_use =</span> <span class="st">&quot;counts&quot;</span>,</a>
<a class="sourceLine" id="cb21-4" title="4">    <span class="dt">marginal_list =</span> PANCREAS_marginal,</a>
<a class="sourceLine" id="cb21-5" title="5">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb21-6" title="6">    <span class="dt">copula =</span> <span class="st">&quot;gaussian&quot;</span>,</a>
<a class="sourceLine" id="cb21-7" title="7">    <span class="dt">n_cores =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb21-8" title="8">    <span class="dt">new_covariate =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb21-9" title="9">    <span class="dt">input_data =</span> PANCREAS_data<span class="op">$</span>dat</a>
<a class="sourceLine" id="cb21-10" title="10">  )</a></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb22-1" title="1">PANCREAS_para &lt;-<span class="st"> </span><span class="kw">extract_para</span>(</a>
<a class="sourceLine" id="cb22-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb22-3" title="3">    <span class="dt">marginal_list =</span> PANCREAS_marginal,</a>
<a class="sourceLine" id="cb22-4" title="4">    <span class="dt">n_cores =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb22-5" title="5">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb22-6" title="6">    <span class="dt">new_covariate =</span> <span class="ot">NULL</span></a>
<a class="sourceLine" id="cb22-7" title="7">  )</a></code></pre></div>
<p>Here, we examine the <code>mean_mat</code>, which is one of the outputs from the previous function <code>extract_para()</code>. For each gene, we calculate the difference in the between the maximum mean parameter and minimum mean parameter across all cells. We select genes which the gene’s mean difference across cells are in the top 50 largest differences. We regard these genes as DE genes(DEG). Then, we manually set the mean parameters of the rest genes to be the same across all cells. We regard all genes with the same mean parameter across cells as non-DE genes.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" title="1">diff &lt;-<span class="st"> </span><span class="kw">apply</span>(PANCREAS_para<span class="op">$</span>mean_mat, <span class="dv">2</span>, <span class="cf">function</span>(x){<span class="kw">max</span>(x)<span class="op">-</span><span class="kw">min</span>(x)})</a>
<a class="sourceLine" id="cb23-2" title="2">diff_ordered &lt;-<span class="st"> </span><span class="kw">order</span>(diff, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb23-3" title="3">diff &lt;-<span class="st"> </span>diff[diff_ordered]</a>
<a class="sourceLine" id="cb23-4" title="4">num_de &lt;-<span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb23-5" title="5">de_idx &lt;-<span class="st"> </span><span class="kw">names</span>(diff[<span class="dv">1</span><span class="op">:</span>num_de])</a>
<a class="sourceLine" id="cb23-6" title="6">non_de_idx &lt;-<span class="st"> </span><span class="kw">names</span>(diff[<span class="op">-</span>(<span class="dv">1</span><span class="op">:</span>num_de)])</a>
<a class="sourceLine" id="cb23-7" title="7">non_de_mat &lt;-<span class="st"> </span><span class="kw">apply</span>(PANCREAS_para<span class="op">$</span>mean_mat[,non_de_idx], <span class="dv">2</span>, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb23-8" title="8">  avg &lt;-<span class="st"> </span><span class="kw">mean</span>(x)</a>
<a class="sourceLine" id="cb23-9" title="9">  new_mean &lt;-<span class="st"> </span><span class="kw">rep</span>(avg, <span class="kw">length</span>(x))</a>
<a class="sourceLine" id="cb23-10" title="10">  <span class="kw">return</span>(new_mean)</a>
<a class="sourceLine" id="cb23-11" title="11">})</a>
<a class="sourceLine" id="cb23-12" title="12">PANCREAS_para<span class="op">$</span>mean_mat[,non_de_idx] &lt;-<span class="st"> </span>non_de_mat</a></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb24-1" title="1"><span class="kw">set.seed</span>(<span class="dv">123</span>)</a>
<a class="sourceLine" id="cb24-2" title="2"> PANCREAS_newcount &lt;-<span class="st"> </span><span class="kw">simu_new</span>(</a>
<a class="sourceLine" id="cb24-3" title="3">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb24-4" title="4">    <span class="dt">mean_mat =</span> PANCREAS_para<span class="op">$</span>mean_mat,</a>
<a class="sourceLine" id="cb24-5" title="5">    <span class="dt">sigma_mat =</span> PANCREAS_para<span class="op">$</span>sigma_mat,</a>
<a class="sourceLine" id="cb24-6" title="6">    <span class="dt">zero_mat =</span> PANCREAS_para<span class="op">$</span>zero_mat,</a>
<a class="sourceLine" id="cb24-7" title="7">    <span class="dt">quantile_mat =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb24-8" title="8">    <span class="dt">copula_list =</span> PANCREAS_copula<span class="op">$</span>copula_list,</a>
<a class="sourceLine" id="cb24-9" title="9">    <span class="dt">n_cores =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb24-10" title="10">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb24-11" title="11">    <span class="dt">input_data =</span> PANCREAS_data<span class="op">$</span>dat,</a>
<a class="sourceLine" id="cb24-12" title="12">    <span class="dt">new_covariate =</span> PANCREAS_data<span class="op">$</span>new_covariate,</a>
<a class="sourceLine" id="cb24-13" title="13">    <span class="dt">important_feature =</span> PANCREAS_copula<span class="op">$</span>important_feature</a>
<a class="sourceLine" id="cb24-14" title="14">  )</a></code></pre></div>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" title="1"><span class="kw">logcounts</span>(example_sce) &lt;-<span class="st"> </span><span class="kw">log1p</span>(<span class="kw">counts</span>(example_sce))</a>
<a class="sourceLine" id="cb25-2" title="2">simu_sce &lt;-<span class="st"> </span>example_sce</a>
<a class="sourceLine" id="cb25-3" title="3"><span class="kw">counts</span>(simu_sce) &lt;-<span class="st">  </span>PANCREAS_newcount</a>
<a class="sourceLine" id="cb25-4" title="4"><span class="kw">logcounts</span>(simu_sce) &lt;-<span class="st"> </span><span class="kw">log1p</span>(<span class="kw">counts</span>(simu_sce))</a></code></pre></div>
</div>
<div id="de-genes-identification" class="section level3">
<h3>DE genes identification</h3>
<p>Now, we use the simulated data to benchmark the performance of two DEG identification methods: <a href="https://www.nature.com/articles/nmeth.4636">Monocle3</a>, <a href="https://www.nature.com/articles/s41467-020-14766-3">tradeSeq</a>, <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-020-3506-x">NBAMseq</a>, and <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02341-y">PseudotimeDE</a>. The p-values from the two methods after Benjamini-Hochberg(BH) corretion will be stored in <code>qvals</code> in the following code.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb26-1" title="1">qvals &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">ncol =</span> <span class="dv">4</span>, <span class="dt">nrow =</span> <span class="kw">dim</span>(simu_sce)[<span class="dv">1</span>])</a>
<a class="sourceLine" id="cb26-2" title="2"><span class="kw">colnames</span>(qvals) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Monocle3-DE&quot;</span>,<span class="st">&quot;tradeSeq&quot;</span>,<span class="st">&quot;NBAMseq&quot;</span>,<span class="st">&quot;PseudotimeDE&quot;</span>)</a>
<a class="sourceLine" id="cb26-3" title="3"><span class="kw">rownames</span>(qvals) &lt;-<span class="st"> </span><span class="kw">rownames</span>(simu_sce)</a></code></pre></div>
<div id="monocle3" class="section level4">
<h4>Monocle3</h4>
<p>We follow the <a href="https://cole-trapnell-lab.github.io/monocle3/docs/differential/">tutorial</a> from <code>Monocle3</code> to conduct the DE test and obtain the p-values after BH correction.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" title="1">rowdata &lt;-<span class="st"> </span><span class="kw">rowData</span>(simu_sce)</a>
<a class="sourceLine" id="cb27-2" title="2">rowdata<span class="op">$</span>gene_short_name &lt;-<span class="st"> </span><span class="kw">rownames</span>(<span class="kw">rowData</span>(simu_sce))</a>
<a class="sourceLine" id="cb27-3" title="3">coldata &lt;-<span class="st"> </span><span class="kw">colData</span>(simu_sce)</a>
<a class="sourceLine" id="cb27-4" title="4">cds &lt;-<span class="st"> </span><span class="kw">new_cell_data_set</span>(<span class="kw">counts</span>(simu_sce),</a>
<a class="sourceLine" id="cb27-5" title="5">                           <span class="dt">cell_metadata =</span> coldata,</a>
<a class="sourceLine" id="cb27-6" title="6">                           <span class="dt">gene_metadata =</span> rowdata)</a>
<a class="sourceLine" id="cb27-7" title="7">gene_fits &lt;-<span class="st"> </span><span class="kw">fit_models</span>(cds, <span class="dt">model_formula_str =</span> <span class="st">&quot;~pseudotime&quot;</span>)</a>
<a class="sourceLine" id="cb27-8" title="8">fit_coefs &lt;-<span class="st"> </span><span class="kw">coefficient_table</span>(gene_fits)</a>
<a class="sourceLine" id="cb27-9" title="9">fit &lt;-<span class="st"> </span>fit_coefs <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">filter</span>(term <span class="op">==</span><span class="st"> &quot;pseudotime&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">select</span>(p_value, q_value) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">gene =</span> <span class="kw">rownames</span>(simu_sce))</a>
<a class="sourceLine" id="cb27-10" title="10">fit</a>
<a class="sourceLine" id="cb27-11" title="11">qvals[fit<span class="op">$</span>gene,<span class="st">&quot;Monocle3-DE&quot;</span>] &lt;-<span class="st"> </span>fit<span class="op">$</span>q_value</a></code></pre></div>
</div>
<div id="tradeseq" class="section level4">
<h4>tradeSeq</h4>
<p>We follow the <a href="https://statomics.github.io/tradeSeq/articles/tradeSeq.html">tutorial</a> from <code>tradeSeq</code> to conduct the DE test and obtain the p-values after BH correction.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" title="1">pseudo &lt;-<span class="st"> </span><span class="kw">colData</span>(simu_sce)<span class="op">$</span>pseudotime</a>
<a class="sourceLine" id="cb28-2" title="2">icMat &lt;-<span class="st"> </span><span class="kw">evaluateK</span>(<span class="dt">counts =</span> <span class="kw">counts</span>(simu_sce), <span class="dt">pseudotime  =</span> pseudo, <span class="dt">cellWeights =</span> <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">dim</span>(simu_sce)[<span class="dv">2</span>]), <span class="dt">k =</span> <span class="dv">3</span><span class="op">:</span><span class="dv">10</span>, <span class="dt">nGenes =</span> <span class="dv">100</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>, <span class="dt">plot =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-30-1.png" width="700" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb29-1" title="1">res_tradeSeq &lt;-<span class="st"> </span><span class="kw">fitGAM</span>(<span class="dt">counts =</span> <span class="kw">as.matrix</span>(<span class="kw">assays</span>(simu_sce)<span class="op">$</span>counts), <span class="dt">pseudotime =</span> pseudo, <span class="dt">cellWeights =</span> <span class="kw">rep</span>(<span class="dv">1</span>, <span class="kw">length</span>(pseudo)),<span class="dt">nknots =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb29-2" title="2">assoRes &lt;-<span class="st"> </span><span class="kw">associationTest</span>(res_tradeSeq, <span class="dt">lineages =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb29-3" title="3">assoRes &lt;-<span class="st"> </span>assoRes <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">as_tibble</span>(<span class="dt">rownames =</span> <span class="st">&quot;gene&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span>dplyr<span class="op">::</span><span class="kw">mutate</span>(<span class="dt">qvalue =</span> <span class="kw">p.adjust</span>(pvalue, <span class="dt">method =</span> <span class="st">&quot;BH&quot;</span>))</a>
<a class="sourceLine" id="cb29-4" title="4">qvals[assoRes<span class="op">$</span>gene,<span class="st">&quot;tradeSeq&quot;</span>] &lt;-<span class="st"> </span>assoRes<span class="op">$</span>qvalue</a></code></pre></div>
</div>
<div id="nbamseq" class="section level4">
<h4>NBAMseq</h4>
<p>We follow the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/NBAMSeq/inst/doc/NBAMSeq-vignette.html">tutorial</a> from <code>NBAMseq</code> to conduct the DE test and obtain the p-values after BH correction.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" title="1">countdata &lt;-<span class="st"> </span><span class="kw">counts</span>(simu_sce)</a>
<a class="sourceLine" id="cb30-2" title="2">coldata &lt;-<span class="st"> </span><span class="kw">colData</span>(simu_sce)</a>
<a class="sourceLine" id="cb30-3" title="3">design =<span class="st"> </span><span class="er">~</span><span class="kw">s</span>(pseudotime)</a>
<a class="sourceLine" id="cb30-4" title="4">gsd =<span class="st"> </span><span class="kw">NBAMSeqDataSet</span>(<span class="dt">countData =</span> countdata, <span class="dt">colData =</span> coldata, <span class="dt">design =</span> design)</a>
<a class="sourceLine" id="cb30-5" title="5">gsd =<span class="st"> </span><span class="kw">NBAMSeq</span>(gsd) </a>
<a class="sourceLine" id="cb30-6" title="6">res1 =<span class="st"> </span><span class="kw">results</span>(gsd, <span class="dt">name =</span> <span class="st">&quot;pseudotime&quot;</span>)</a>
<a class="sourceLine" id="cb30-7" title="7"><span class="kw">head</span>(res1)</a>
<a class="sourceLine" id="cb30-8" title="8">qvals[<span class="kw">rownames</span>(res1),<span class="st">&quot;NBAMseq&quot;</span>] =<span class="st"> </span>res1<span class="op">$</span>padj</a></code></pre></div>
</div>
<div id="pseudotimede" class="section level4">
<h4>PseudotimeDE</h4>
<p>We follow the <a href="https://htmlpreview.github.io/?https://rpubs.com/dongyuansong/842884">tutorial</a> from <code>PseudotimeDE</code> to conduct the DE test and obtain the p-values after BH correction. Here to save computational time, we use the fix-mode of PseudotimeDE.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb31-1" title="1">ori_tbl &lt;-<span class="st"> </span><span class="kw">tibble</span>(<span class="dt">cell =</span> <span class="kw">colnames</span>(simu_sce), <span class="dt">pseudotime =</span> <span class="kw">colData</span>(simu_sce)<span class="op">$</span>pseudotime)</a>
<a class="sourceLine" id="cb31-2" title="2">res &lt;-<span class="st"> </span><span class="kw">runPseudotimeDE</span>(<span class="dt">gene.vec =</span> <span class="kw">rownames</span>(simu_sce),</a>
<a class="sourceLine" id="cb31-3" title="3">                                     <span class="dt">ori.tbl =</span> ori_tbl,</a>
<a class="sourceLine" id="cb31-4" title="4">                                     <span class="dt">sub.tbl =</span> <span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb31-5" title="5">                                     <span class="dt">mat =</span> simu_sce, </a>
<a class="sourceLine" id="cb31-6" title="6">                                     <span class="dt">model =</span> <span class="st">&quot;nb&quot;</span>)</a>
<a class="sourceLine" id="cb31-7" title="7">qvals[res<span class="op">$</span>gene,<span class="st">&quot;PseudotimeDE&quot;</span>] &lt;-<span class="st"> </span><span class="kw">p.adjust</span>(res<span class="op">$</span>fix.pv, <span class="dt">method =</span> <span class="st">&quot;BH&quot;</span>, <span class="kw">length</span>(res<span class="op">$</span>gene))</a></code></pre></div>
<p>Since <code>tradeSeq</code>’s result contains some NA, we convert the NA to 1 first.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" title="1">qvals[<span class="kw">is.na</span>(qvals)] &lt;-<span class="st"> </span><span class="dv">1</span></a>
<a class="sourceLine" id="cb32-2" title="2"><span class="kw">print</span>(<span class="kw">head</span>(qvals))</a></code></pre></div>
<p>Since we manually created non-DE genes in the <code>extra_para()</code> step, now we can calculate the actual false discovery proportion(FDP) and power of the DE tests we conducted above with various target FDR threshold.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb33-1" title="1">test =<span class="st"> </span><span class="kw">colnames</span>(qvals)</a>
<a class="sourceLine" id="cb33-2" title="2">targetFDR &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">seq</span>(<span class="fl">0.01</span>,<span class="fl">0.1</span>,<span class="dt">by=</span><span class="fl">0.01</span>),<span class="kw">seq</span>(<span class="fl">0.2</span>,<span class="fl">0.5</span>,<span class="dt">by=</span><span class="fl">0.1</span>))</a>
<a class="sourceLine" id="cb33-3" title="3">de &lt;-<span class="st"> </span><span class="kw">colnames</span>(PANCREAS_para<span class="op">$</span>mean_mat[,de_idx])</a>
<a class="sourceLine" id="cb33-4" title="4">fdp_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(targetFDR), <span class="dt">ncol =</span> <span class="kw">length</span>(test))</a>
<a class="sourceLine" id="cb33-5" title="5"><span class="kw">colnames</span>(fdp_mat) &lt;-<span class="st"> </span>test</a>
<a class="sourceLine" id="cb33-6" title="6"><span class="kw">rownames</span>(fdp_mat) &lt;-<span class="st"> </span>targetFDR</a>
<a class="sourceLine" id="cb33-7" title="7">power_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(targetFDR), <span class="dt">ncol =</span> <span class="kw">length</span>(test))</a>
<a class="sourceLine" id="cb33-8" title="8"><span class="kw">colnames</span>(power_mat) &lt;-<span class="st"> </span>test</a>
<a class="sourceLine" id="cb33-9" title="9"><span class="kw">rownames</span>(power_mat) &lt;-<span class="st"> </span>targetFDR</a>
<a class="sourceLine" id="cb33-10" title="10"><span class="cf">for</span> (t <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(test)) {</a>
<a class="sourceLine" id="cb33-11" title="11">  curr_p =<span class="st"> </span>qvals[,t]</a>
<a class="sourceLine" id="cb33-12" title="12">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(targetFDR)) {</a>
<a class="sourceLine" id="cb33-13" title="13">    thre &lt;-<span class="st"> </span>targetFDR[i]</a>
<a class="sourceLine" id="cb33-14" title="14">    discovery &lt;-<span class="st"> </span><span class="kw">which</span>(curr_p <span class="op">&lt;=</span><span class="st"> </span>thre)</a>
<a class="sourceLine" id="cb33-15" title="15">    tp &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(<span class="kw">names</span>(discovery),de))</a>
<a class="sourceLine" id="cb33-16" title="16">    <span class="cf">if</span>(<span class="kw">length</span>(discovery) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>){</a>
<a class="sourceLine" id="cb33-17" title="17">      fdp &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb33-18" title="18">    }<span class="cf">else</span>{</a>
<a class="sourceLine" id="cb33-19" title="19">      fdp &lt;-<span class="st"> </span>(<span class="kw">length</span>(discovery) <span class="op">-</span><span class="st"> </span>tp)<span class="op">/</span><span class="kw">length</span>(discovery)</a>
<a class="sourceLine" id="cb33-20" title="20">    }</a>
<a class="sourceLine" id="cb33-21" title="21">  </a>
<a class="sourceLine" id="cb33-22" title="22">    power &lt;-<span class="st"> </span>tp<span class="op">/</span><span class="kw">length</span>(de)</a>
<a class="sourceLine" id="cb33-23" title="23">    fdp_mat[i, t] &lt;-<span class="st"> </span>fdp</a>
<a class="sourceLine" id="cb33-24" title="24">    power_mat[i,t] &lt;-<span class="st"> </span>power</a>
<a class="sourceLine" id="cb33-25" title="25">  }</a>
<a class="sourceLine" id="cb33-26" title="26">}</a></code></pre></div>
<p>Lastly, we visualize the Target FDR vs Actual FDP and Target FDR vs Power below.</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" title="1">fdp_long &lt;-<span class="st"> </span><span class="kw">melt</span>(fdp_mat)</a>
<a class="sourceLine" id="cb34-2" title="2"><span class="kw">colnames</span>(fdp_long) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Target FDR&quot;</span>,<span class="st">&quot;test_method&quot;</span>,<span class="st">&quot;Actual FDP&quot;</span>)</a>
<a class="sourceLine" id="cb34-3" title="3">fdp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(fdp_long) <span class="op">+</span></a>
<a class="sourceLine" id="cb34-4" title="4"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span><span class="st">`</span><span class="dt">Actual FDP</span><span class="st">`</span>,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb34-5" title="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span><span class="st">`</span><span class="dt">Actual FDP</span><span class="st">`</span>,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb34-6" title="6"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>,<span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>,<span class="dt">color=</span><span class="st">&quot;grey&quot;</span>)<span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb34-7" title="7"><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">expand_limits</span>(<span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</a>
<a class="sourceLine" id="cb34-8" title="8">fdp_plot</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-36-1.png" width="700" /></p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb35-1" title="1">power_long &lt;-<span class="st"> </span><span class="kw">melt</span>(power_mat)</a>
<a class="sourceLine" id="cb35-2" title="2"><span class="kw">colnames</span>(power_long) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Target FDR&quot;</span>,<span class="st">&quot;test_method&quot;</span>,<span class="st">&quot;Power&quot;</span>)</a>
<a class="sourceLine" id="cb35-3" title="3">power_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(power_long) <span class="op">+</span></a>
<a class="sourceLine" id="cb35-4" title="4"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span>Power,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb35-5" title="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span>Power,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb35-6" title="6"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb35-7" title="7">power_plot</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-37-1.png" width="700" /></p>
</div>
</div>
</div>
<div id="identication-of-spatially-variable-genes-svg-in-spatial-transcriptomic-data" class="section level2">
<h2>Identication of Spatially Variable Genes (SVG) in spatial transcriptomic data</h2>
<div id="read-in-the-reference-data-2" class="section level3">
<h3>Read in the reference data</h3>
<p>The raw data is from the <a href="https://satijalab.org/seurat/articles/spatial_vignette.html">Seurat</a>, which is a dataset generated with the Visium technology from 10x Genomics. We pre-select the top spatial variable genes.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" title="1">example_sce &lt;-<span class="st"> </span><span class="kw">readRDS</span>((<span class="kw">url</span>(<span class="st">&quot;https://www.dropbox.com/s/v1uazcl2glirmfe/VISIUM_sce.rds?raw=1&quot;</span>)))</a>
<a class="sourceLine" id="cb36-2" title="2"><span class="kw">print</span>(example_sce)</a>
<a class="sourceLine" id="cb36-3" title="3"><span class="co">#&gt; class: SingleCellExperiment </span></a>
<a class="sourceLine" id="cb36-4" title="4"><span class="co">#&gt; dim: 1000 2696 </span></a>
<a class="sourceLine" id="cb36-5" title="5"><span class="co">#&gt; metadata(0):</span></a>
<a class="sourceLine" id="cb36-6" title="6"><span class="co">#&gt; assays(2): counts logcounts</span></a>
<a class="sourceLine" id="cb36-7" title="7"><span class="co">#&gt; rownames(1000): Calb2 Gng4 ... Fndc5 Gda</span></a>
<a class="sourceLine" id="cb36-8" title="8"><span class="co">#&gt; rowData names(0):</span></a>
<a class="sourceLine" id="cb36-9" title="9"><span class="co">#&gt; colnames(2696): AAACAAGTATCTCCCA-1 AAACACCAATAACTGC-1 ...</span></a>
<a class="sourceLine" id="cb36-10" title="10"><span class="co">#&gt;   TTGTTTCACATCCAGG-1 TTGTTTCCATACAACT-1</span></a>
<a class="sourceLine" id="cb36-11" title="11"><span class="co">#&gt; colData names(12): orig.ident nCount_Spatial ... spatial2 cell_type</span></a>
<a class="sourceLine" id="cb36-12" title="12"><span class="co">#&gt; reducedDimNames(0):</span></a>
<a class="sourceLine" id="cb36-13" title="13"><span class="co">#&gt; mainExpName: NULL</span></a>
<a class="sourceLine" id="cb36-14" title="14"><span class="co">#&gt; altExpNames(0):</span></a></code></pre></div>
<p>To save time, we subset the top 200 genes.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb37-1" title="1">example_sce &lt;-<span class="st"> </span>example_sce[<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>, ]</a></code></pre></div>
<p>We remove the mitochondrial genes since this is a prepreocessing step in SPARK-X, which is one of the method we will use to identify spatially varaible genes.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" title="1">mt_idx&lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;mt-&quot;</span>,<span class="kw">rownames</span>(example_sce))</a>
<a class="sourceLine" id="cb38-2" title="2"><span class="cf">if</span>(<span class="kw">length</span>(mt_idx)<span class="op">!=</span><span class="dv">0</span>){</a>
<a class="sourceLine" id="cb38-3" title="3">    example_sce   &lt;-<span class="st"> </span>example_sce[<span class="op">-</span>mt_idx,]</a>
<a class="sourceLine" id="cb38-4" title="4">}</a></code></pre></div>
</div>
<div id="simulation-2" class="section level3">
<h3>Simulation</h3>
<p>We use the step-by-step functions instead of the one-shot function to generate synthetic data since these step-by-step functions allow us to alter estimated parameters and generate new data based on our desired parameters.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb39-1" title="1">example_data &lt;-<span class="st"> </span><span class="kw">construct_data</span>(</a>
<a class="sourceLine" id="cb39-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb39-3" title="3">    <span class="dt">assay_use =</span> <span class="st">&quot;counts&quot;</span>,</a>
<a class="sourceLine" id="cb39-4" title="4">    <span class="dt">celltype =</span> <span class="st">&quot;cell_type&quot;</span>,</a>
<a class="sourceLine" id="cb39-5" title="5">    <span class="dt">pseudotime =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb39-6" title="6">    <span class="dt">spatial =</span> <span class="kw">c</span>(<span class="st">&quot;spatial1&quot;</span>, <span class="st">&quot;spatial2&quot;</span>),</a>
<a class="sourceLine" id="cb39-7" title="7">    <span class="dt">other_covariates =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb39-8" title="8">    <span class="dt">corr_by =</span> <span class="st">&quot;1&quot;</span></a>
<a class="sourceLine" id="cb39-9" title="9">  )</a></code></pre></div>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" title="1">example_marginal &lt;-<span class="st"> </span><span class="kw">fit_marginal</span>(</a>
<a class="sourceLine" id="cb40-2" title="2">    <span class="dt">data =</span> example_data,</a>
<a class="sourceLine" id="cb40-3" title="3">    <span class="dt">predictor =</span> <span class="st">&quot;gene&quot;</span>,</a>
<a class="sourceLine" id="cb40-4" title="4">    <span class="dt">mu_formula =</span> <span class="st">&quot;s(spatial1, spatial2, bs = &#39;gp&#39;, k= 50)&quot;</span>,</a>
<a class="sourceLine" id="cb40-5" title="5">    <span class="dt">sigma_formula =</span> <span class="st">&quot;1&quot;</span>,</a>
<a class="sourceLine" id="cb40-6" title="6">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb40-7" title="7">    <span class="dt">n_cores =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb40-8" title="8">    <span class="dt">usebam =</span> <span class="ot">FALSE</span></a>
<a class="sourceLine" id="cb40-9" title="9">  )</a></code></pre></div>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" title="1">example_copula &lt;-<span class="st"> </span><span class="kw">fit_copula</span>(</a>
<a class="sourceLine" id="cb41-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb41-3" title="3">    <span class="dt">assay_use =</span> <span class="st">&quot;counts&quot;</span>,</a>
<a class="sourceLine" id="cb41-4" title="4">    <span class="dt">marginal_list =</span> example_marginal,</a>
<a class="sourceLine" id="cb41-5" title="5">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb41-6" title="6">    <span class="dt">copula =</span> <span class="st">&quot;gaussian&quot;</span>,</a>
<a class="sourceLine" id="cb41-7" title="7">    <span class="dt">n_cores =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb41-8" title="8">    <span class="dt">new_covariate =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb41-9" title="9">    <span class="dt">input_data =</span> example_data<span class="op">$</span>dat</a>
<a class="sourceLine" id="cb41-10" title="10">  )</a></code></pre></div>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" title="1">example_para &lt;-<span class="st"> </span><span class="kw">extract_para</span>(</a>
<a class="sourceLine" id="cb42-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb42-3" title="3">    <span class="dt">marginal_list =</span> example_marginal,</a>
<a class="sourceLine" id="cb42-4" title="4">    <span class="dt">n_cores =</span> <span class="dv">2</span>,</a>
<a class="sourceLine" id="cb42-5" title="5">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb42-6" title="6">    <span class="dt">new_covariate =</span> <span class="ot">NULL</span></a>
<a class="sourceLine" id="cb42-7" title="7">  )</a></code></pre></div>
<p>Here, we examine the <code>mean_mat</code>, which is one of the outputs from the previous function <code>extract_para()</code>. For each gene, we calculate the deviance explained by spatial locations in each regression model, and select the top 50. We regard these genes as spatially variable genes (SVGs). Then, we manually set the mean parameters of the rest genes to be the same across all cells. We regard all genes with the same mean parameter across cells as non-SVG genes.</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb43-1" title="1">dev_explain &lt;-<span class="st"> </span><span class="kw">sapply</span>(example_marginal, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb43-2" title="2">  sum =<span class="st"> </span><span class="kw">summary</span>(x)</a>
<a class="sourceLine" id="cb43-3" title="3">  <span class="kw">return</span>(sum<span class="op">$</span>dev.expl)</a>
<a class="sourceLine" id="cb43-4" title="4">})</a>
<a class="sourceLine" id="cb43-5" title="5">dev_ordered &lt;-<span class="st"> </span><span class="kw">order</span>(dev_explain, <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb43-6" title="6">num_de &lt;-<span class="st"> </span><span class="dv">50</span></a>
<a class="sourceLine" id="cb43-7" title="7">ordered &lt;-<span class="st"> </span>dev_explain[dev_ordered]</a>
<a class="sourceLine" id="cb43-8" title="8">de_idx &lt;-<span class="st"> </span><span class="kw">names</span>(ordered)[<span class="dv">1</span><span class="op">:</span>num_de]</a>
<a class="sourceLine" id="cb43-9" title="9">non_de_idx &lt;-<span class="st"> </span><span class="kw">names</span>(ordered)[<span class="op">-</span>(<span class="dv">1</span><span class="op">:</span>num_de)]</a>
<a class="sourceLine" id="cb43-10" title="10">non_de_mat &lt;-<span class="st"> </span><span class="kw">apply</span>(example_para<span class="op">$</span>mean_mat[,non_de_idx], <span class="dv">2</span>, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb43-11" title="11">  avg &lt;-<span class="st"> </span>(<span class="kw">max</span>(x)<span class="op">+</span><span class="kw">min</span>(x))<span class="op">/</span><span class="dv">2</span></a>
<a class="sourceLine" id="cb43-12" title="12">  new_mean &lt;-<span class="st"> </span><span class="kw">rep</span>(avg, <span class="kw">length</span>(x))</a>
<a class="sourceLine" id="cb43-13" title="13">  <span class="kw">return</span>(new_mean)</a>
<a class="sourceLine" id="cb43-14" title="14">})</a>
<a class="sourceLine" id="cb43-15" title="15">example_para<span class="op">$</span>mean_mat[,non_de_idx] &lt;-<span class="st"> </span>non_de_mat</a></code></pre></div>
<p>Another way to select SVG based on Moran’s I.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" title="1"><span class="co"># num_de &lt;- 50</span></a>
<a class="sourceLine" id="cb44-2" title="2"><span class="co"># loc = colData(simu_sce)[,c(&quot;spatial1&quot;,&quot;spatial2&quot;)]</span></a>
<a class="sourceLine" id="cb44-3" title="3"><span class="co"># features = FindSpatiallyVariableFeatures(counts(simu_sce), spatial.location = loc, selection.method = &quot;moransi&quot;,nfeatures = num_de)</span></a>
<a class="sourceLine" id="cb44-4" title="4"><span class="co"># top.features = features[order(features$p.value),]</span></a>
<a class="sourceLine" id="cb44-5" title="5"><span class="co"># top.features= rownames(top.features[1:num_de,])</span></a>
<a class="sourceLine" id="cb44-6" title="6"><span class="co"># de_idx &lt;- which(rownames(simu_sce) %in% top.features)</span></a>
<a class="sourceLine" id="cb44-7" title="7"><span class="co"># non_de_idx &lt;-which(!rownames(simu_sce) %in% top.features)</span></a>
<a class="sourceLine" id="cb44-8" title="8"><span class="co"># non_de_mat &lt;- apply(example_para$mean_mat[,non_de_idx], 2, function(x){</span></a>
<a class="sourceLine" id="cb44-9" title="9"><span class="co">#   avg &lt;- (max(x)+min(x))/2</span></a>
<a class="sourceLine" id="cb44-10" title="10"><span class="co">#   new_mean &lt;- rep(avg, length(x))</span></a>
<a class="sourceLine" id="cb44-11" title="11"><span class="co">#   return(new_mean)</span></a>
<a class="sourceLine" id="cb44-12" title="12"><span class="co"># })</span></a>
<a class="sourceLine" id="cb44-13" title="13"><span class="co"># example_para$mean_mat[,non_de_idx] &lt;- non_de_mat</span></a></code></pre></div>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" title="1"> example_newcount &lt;-<span class="st"> </span><span class="kw">simu_new</span>(</a>
<a class="sourceLine" id="cb45-2" title="2">    <span class="dt">sce =</span> example_sce,</a>
<a class="sourceLine" id="cb45-3" title="3">    <span class="dt">mean_mat =</span> example_para<span class="op">$</span>mean_mat,</a>
<a class="sourceLine" id="cb45-4" title="4">    <span class="dt">sigma_mat =</span> example_para<span class="op">$</span>sigma_mat,</a>
<a class="sourceLine" id="cb45-5" title="5">    <span class="dt">zero_mat =</span> example_para<span class="op">$</span>zero_mat,</a>
<a class="sourceLine" id="cb45-6" title="6">    <span class="dt">quantile_mat =</span> <span class="ot">NULL</span>,</a>
<a class="sourceLine" id="cb45-7" title="7">    <span class="dt">copula_list =</span> example_copula<span class="op">$</span>copula_list,</a>
<a class="sourceLine" id="cb45-8" title="8">    <span class="dt">n_cores =</span> <span class="dv">1</span>,</a>
<a class="sourceLine" id="cb45-9" title="9">    <span class="dt">family_use =</span> <span class="st">&quot;nb&quot;</span>,</a>
<a class="sourceLine" id="cb45-10" title="10">    <span class="dt">input_data =</span> example_data<span class="op">$</span>dat,</a>
<a class="sourceLine" id="cb45-11" title="11">    <span class="dt">new_covariate =</span> example_data<span class="op">$</span>new_covariate,</a>
<a class="sourceLine" id="cb45-12" title="12">    <span class="dt">important_feature =</span> <span class="kw">rep</span>(<span class="ot">TRUE</span>, <span class="kw">dim</span>(example_sce)[<span class="dv">1</span>])</a>
<a class="sourceLine" id="cb45-13" title="13">  )</a></code></pre></div>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" title="1"><span class="kw">logcounts</span>(example_sce) &lt;-<span class="st"> </span><span class="kw">log1p</span>(<span class="kw">counts</span>(example_sce))</a>
<a class="sourceLine" id="cb46-2" title="2">simu_sce &lt;-<span class="st"> </span>example_sce</a>
<a class="sourceLine" id="cb46-3" title="3"><span class="kw">counts</span>(simu_sce) &lt;-<span class="st"> </span>example_newcount</a>
<a class="sourceLine" id="cb46-4" title="4"><span class="kw">logcounts</span>(simu_sce) &lt;-<span class="st"> </span><span class="kw">log1p</span>(<span class="kw">counts</span>(simu_sce))</a></code></pre></div>
<p>We rescale the log count of 5 synthetic SVG here to the 0 to 1 scale and visualize them below.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" title="1">de_genes =<span class="st"> </span>de_idx[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb47-2" title="2">loc =<span class="st"> </span><span class="kw">colData</span>(simu_sce)[,<span class="kw">c</span>(<span class="st">&quot;spatial1&quot;</span>,<span class="st">&quot;spatial2&quot;</span>)]</a>
<a class="sourceLine" id="cb47-3" title="3">expre =<span class="st"> </span><span class="kw">lapply</span>(de_genes, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb47-4" title="4">    curr =<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">counts</span>(simu_sce)[x,])</a>
<a class="sourceLine" id="cb47-5" title="5">    curr =<span class="st"> </span><span class="kw">log1p</span>(curr)</a>
<a class="sourceLine" id="cb47-6" title="6">    <span class="kw">return</span>(<span class="kw">rescale</span>(curr))</a>
<a class="sourceLine" id="cb47-7" title="7">  })</a>
<a class="sourceLine" id="cb47-8" title="8">long =<span class="st"> </span><span class="kw">do.call</span>(rbind, expre)</a>
<a class="sourceLine" id="cb47-9" title="9">long =<span class="st"> </span><span class="kw">as.data.frame</span>(long)</a>
<a class="sourceLine" id="cb47-10" title="10"><span class="kw">colnames</span>(long) &lt;-<span class="st"> &quot;Expression&quot;</span></a>
<a class="sourceLine" id="cb47-11" title="11">long<span class="op">$</span>gene =<span class="st"> </span><span class="kw">do.call</span>(c, <span class="kw">lapply</span>(de_genes, <span class="cf">function</span>(x){<span class="kw">rep</span>(x,<span class="kw">dim</span>(expre[[<span class="dv">1</span>]])[<span class="dv">1</span>])}))</a>
<a class="sourceLine" id="cb47-12" title="12">long<span class="op">$</span>x =<span class="st"> </span><span class="kw">rep</span>(loc[,<span class="dv">1</span>],<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb47-13" title="13">long<span class="op">$</span>y =<span class="st"> </span><span class="kw">rep</span>(loc[,<span class="dv">2</span>],<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb47-14" title="14"><span class="kw">as_tibble</span>(long, <span class="dt">rownames =</span> <span class="st">&quot;Cell&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> Expression)) <span class="op">+</span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.1</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>gene)<span class="op">+</span><span class="st"> </span><span class="kw">scale_colour_gradientn</span>(<span class="dt">colors =</span> <span class="kw">viridis_pal</span>(<span class="dt">option =</span> <span class="st">&quot;magma&quot;</span>)(<span class="dv">10</span>), <span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">coord_fixed</span>(<span class="dt">ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">45</span>))</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-49-1.png" width="700" /></p>
<p>We also rescale the log count of 5 synthetic non-SVG here to the 0 to 1 scale and visualize them below.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" title="1">non_de_genes =<span class="st"> </span>non_de_idx[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>]</a>
<a class="sourceLine" id="cb48-2" title="2">loc =<span class="st"> </span><span class="kw">colData</span>(simu_sce)[,<span class="kw">c</span>(<span class="st">&quot;spatial1&quot;</span>,<span class="st">&quot;spatial2&quot;</span>)]</a>
<a class="sourceLine" id="cb48-3" title="3">expre =<span class="st"> </span><span class="kw">lapply</span>(non_de_genes, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb48-4" title="4">    curr =<span class="st"> </span><span class="kw">as.matrix</span>(<span class="kw">counts</span>(simu_sce)[x,])</a>
<a class="sourceLine" id="cb48-5" title="5">    curr =<span class="st"> </span><span class="kw">log1p</span>(curr)</a>
<a class="sourceLine" id="cb48-6" title="6">    <span class="kw">return</span>(<span class="kw">rescale</span>(curr))</a>
<a class="sourceLine" id="cb48-7" title="7">  })</a>
<a class="sourceLine" id="cb48-8" title="8">long =<span class="st"> </span><span class="kw">do.call</span>(rbind, expre)</a>
<a class="sourceLine" id="cb48-9" title="9">long =<span class="st"> </span><span class="kw">as.data.frame</span>(long)</a>
<a class="sourceLine" id="cb48-10" title="10"><span class="kw">colnames</span>(long) &lt;-<span class="st"> &quot;Expression&quot;</span></a>
<a class="sourceLine" id="cb48-11" title="11">long<span class="op">$</span>gene =<span class="st"> </span><span class="kw">do.call</span>(c, <span class="kw">lapply</span>(non_de_genes, <span class="cf">function</span>(x){<span class="kw">rep</span>(x,<span class="kw">dim</span>(expre[[<span class="dv">1</span>]])[<span class="dv">1</span>])}))</a>
<a class="sourceLine" id="cb48-12" title="12">long<span class="op">$</span>x =<span class="st"> </span><span class="kw">rep</span>(loc[,<span class="dv">1</span>],<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb48-13" title="13">long<span class="op">$</span>y =<span class="st"> </span><span class="kw">rep</span>(loc[,<span class="dv">2</span>],<span class="dv">5</span>)</a>
<a class="sourceLine" id="cb48-14" title="14"><span class="kw">as_tibble</span>(long, <span class="dt">rownames =</span> <span class="st">&quot;Cell&quot;</span>) <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">ggplot</span>(<span class="kw">aes</span>(<span class="dt">x =</span> x, <span class="dt">y =</span> y, <span class="dt">color =</span> Expression)) <span class="op">+</span><span class="kw">geom_point</span>(<span class="dt">size =</span> <span class="fl">0.1</span>)<span class="op">+</span><span class="kw">facet_grid</span>(<span class="op">~</span>gene)<span class="op">+</span><span class="st"> </span><span class="kw">scale_colour_gradientn</span>(<span class="dt">colors =</span> <span class="kw">viridis_pal</span>(<span class="dt">option =</span> <span class="st">&quot;magma&quot;</span>)(<span class="dv">10</span>), <span class="dt">limits=</span><span class="kw">c</span>(<span class="dv">0</span>, <span class="dv">1</span>)) <span class="op">+</span><span class="st"> </span><span class="kw">coord_fixed</span>(<span class="dt">ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">theme</span>(<span class="dt">axis.text.x =</span> <span class="kw">element_text</span>(<span class="dt">angle =</span> <span class="dv">45</span>))</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-50-1.png" width="700" /> ### SVG identification Now, we use the simulated data to benchmark the performance of two SVG identification methods: <a href="https://www.nature.com/articles/nmeth.4636">spatialDE</a>, <a href="https://www.nature.com/articles/s41592-019-0701-7">SPARK</a>, and <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02404-0">SPARK-X</a>. The p-values from the two methods after Benjamini-Hochberg(BH) corretion will be stored in <code>qvals</code> in the following code.</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" title="1">qvals &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">ncol =</span> <span class="dv">3</span>, <span class="dt">nrow =</span> <span class="kw">dim</span>(simu_sce)[<span class="dv">1</span>])</a>
<a class="sourceLine" id="cb49-2" title="2"><span class="kw">colnames</span>(qvals) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;spatialDE&quot;</span>,<span class="st">&quot;SPARK&quot;</span>,<span class="st">&quot;SPARK-X&quot;</span>)</a>
<a class="sourceLine" id="cb49-3" title="3"><span class="kw">rownames</span>(qvals) &lt;-<span class="st"> </span><span class="kw">rownames</span>(simu_sce)</a></code></pre></div>
<div id="spatialde" class="section level4">
<h4>spatialDE</h4>
<p>We follow the <a href="https://www.bioconductor.org/packages/release/bioc/vignettes/spatialDE/inst/doc/spatialDE.html">tutorial</a> from <code>spatialDE</code> to conduct the hypothesis test and obtain the p-values after BH correction.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" title="1">count &lt;-<span class="st"> </span><span class="kw">counts</span>(simu_sce)</a>
<a class="sourceLine" id="cb50-2" title="2">sample_info &lt;-<span class="st"> </span><span class="kw">colData</span>(simu_sce)[, <span class="kw">c</span>(<span class="st">&quot;spatial1&quot;</span>,<span class="st">&quot;spatial2&quot;</span>)]</a>
<a class="sourceLine" id="cb50-3" title="3">sample_info &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(sample_info)</a>
<a class="sourceLine" id="cb50-4" title="4"><span class="kw">colnames</span>(sample_info) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;x&quot;</span>,<span class="st">&quot;y&quot;</span>)</a>
<a class="sourceLine" id="cb50-5" title="5">count &lt;-<span class="st"> </span>count[<span class="kw">rowSums</span>(count) <span class="op">&gt;=</span><span class="dv">3</span>, ]</a>
<a class="sourceLine" id="cb50-6" title="6">count &lt;-<span class="st"> </span>count[, <span class="kw">row.names</span>(sample_info)]</a>
<a class="sourceLine" id="cb50-7" title="7">sample_info<span class="op">$</span>total_counts &lt;-<span class="st"> </span><span class="kw">colSums</span>(count)</a>
<a class="sourceLine" id="cb50-8" title="8">X &lt;-<span class="st"> </span>sample_info[,<span class="kw">c</span>(<span class="st">&quot;x&quot;</span>,<span class="st">&quot;y&quot;</span>)]</a>
<a class="sourceLine" id="cb50-9" title="9">norm_expr &lt;-<span class="st"> </span><span class="kw">stabilize</span>(count)</a>
<a class="sourceLine" id="cb50-10" title="10">resid_expr &lt;-<span class="st"> </span><span class="kw">regress_out</span>(norm_expr, <span class="dt">sample_info =</span> sample_info)</a>
<a class="sourceLine" id="cb50-11" title="11">results_spatialDE &lt;-<span class="st"> </span>spatialDE<span class="op">::</span><span class="kw">run</span>(resid_expr, <span class="dt">coordinates =</span> X)</a>
<a class="sourceLine" id="cb50-12" title="12"><span class="kw">rownames</span>(results_spatialDE) &lt;-<span class="st"> </span>results_spatialDE<span class="op">$</span>g</a>
<a class="sourceLine" id="cb50-13" title="13">qvals[,<span class="st">&quot;spatialDE&quot;</span>] &lt;-<span class="st"> </span>results_spatialDE[<span class="kw">rownames</span>(simu_sce),<span class="st">&quot;qval&quot;</span>]</a></code></pre></div>
</div>
<div id="spark" class="section level4">
<h4>SPARK</h4>
<p>We then follow the <a href="https://xzhoulab.github.io/SPARK/02_SPARK_Example/">tutorial</a> from <code>SPARK</code> to conduct the hypothesis test and obtain the p-values after BH correction.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" title="1">rawcount &lt;-<span class="st"> </span><span class="kw">counts</span>(simu_sce)</a>
<a class="sourceLine" id="cb51-2" title="2">info &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(simu_sce)[, <span class="kw">c</span>(<span class="st">&quot;spatial1&quot;</span>,<span class="st">&quot;spatial2&quot;</span>)])</a>
<a class="sourceLine" id="cb51-3" title="3"><span class="kw">colnames</span>(info) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;x&quot;</span>,<span class="st">&quot;y&quot;</span>)</a>
<a class="sourceLine" id="cb51-4" title="4">info<span class="op">$</span>total_count &lt;-<span class="st"> </span><span class="kw">apply</span>(rawcount,<span class="dv">2</span>,sum)</a>
<a class="sourceLine" id="cb51-5" title="5"><span class="kw">rownames</span>(info) &lt;-<span class="st"> </span><span class="kw">colnames</span>(rawcount)</a>
<a class="sourceLine" id="cb51-6" title="6">spark &lt;-<span class="st"> </span><span class="kw">CreateSPARKObject</span>(<span class="dt">counts=</span>rawcount, </a>
<a class="sourceLine" id="cb51-7" title="7">                             <span class="dt">location=</span>info[,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>],</a>
<a class="sourceLine" id="cb51-8" title="8">                             <span class="dt">percentage =</span> <span class="dv">0</span>, </a>
<a class="sourceLine" id="cb51-9" title="9">                             <span class="dt">min_total_counts =</span> <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb51-10" title="10">spark<span class="op">@</span>lib_size &lt;-<span class="st"> </span><span class="kw">apply</span>(spark<span class="op">@</span>counts, <span class="dv">2</span>, sum)</a>
<a class="sourceLine" id="cb51-11" title="11">spark &lt;-<span class="st"> </span><span class="kw">spark.vc</span>(spark, </a>
<a class="sourceLine" id="cb51-12" title="12">                   <span class="dt">covariates =</span> <span class="ot">NULL</span>, </a>
<a class="sourceLine" id="cb51-13" title="13">                   <span class="dt">lib_size =</span> spark<span class="op">@</span>lib_size, </a>
<a class="sourceLine" id="cb51-14" title="14">                   <span class="dt">num_core =</span> <span class="dv">5</span>,</a>
<a class="sourceLine" id="cb51-15" title="15">                   <span class="dt">verbose =</span> F)</a>
<a class="sourceLine" id="cb51-16" title="16">spark &lt;-<span class="st"> </span><span class="kw">spark.test</span>(spark, </a>
<a class="sourceLine" id="cb51-17" title="17">                     <span class="dt">check_positive =</span> T, </a>
<a class="sourceLine" id="cb51-18" title="18">                     <span class="dt">verbose =</span> F)</a>
<a class="sourceLine" id="cb51-19" title="19">qvals[,<span class="st">&quot;SPARK&quot;</span>] &lt;-<span class="st"> </span>spark<span class="op">@</span>res_mtest<span class="op">$</span>adjusted_pvalue</a></code></pre></div>
</div>
<div id="spark-x" class="section level4">
<h4>SPARK-X</h4>
<p>We then follow the <a href="https://xzhoulab.github.io/SPARK/02_SPARK_Example/">tutorial</a> from <code>SPARK-X</code> to conduct the hypothesis test and obtain the p-values after BH correction.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" title="1">sp_count &lt;-<span class="st"> </span><span class="kw">counts</span>(simu_sce)</a>
<a class="sourceLine" id="cb52-2" title="2">info &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(<span class="kw">colData</span>(simu_sce)[, <span class="kw">c</span>(<span class="st">&quot;spatial1&quot;</span>,<span class="st">&quot;spatial2&quot;</span>)])</a>
<a class="sourceLine" id="cb52-3" title="3"><span class="kw">colnames</span>(info) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;x&quot;</span>,<span class="st">&quot;y&quot;</span>)</a>
<a class="sourceLine" id="cb52-4" title="4">info<span class="op">$</span>total_count &lt;-<span class="st"> </span><span class="kw">apply</span>(sp_count ,<span class="dv">2</span>,sum)</a>
<a class="sourceLine" id="cb52-5" title="5"><span class="kw">rownames</span>(info) &lt;-<span class="st"> </span><span class="kw">colnames</span>(sp_count )</a>
<a class="sourceLine" id="cb52-6" title="6">location &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(info)</a>
<a class="sourceLine" id="cb52-7" title="7">mt_idx      &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;mt-&quot;</span>,<span class="kw">rownames</span>(sp_count))</a>
<a class="sourceLine" id="cb52-8" title="8"><span class="cf">if</span>(<span class="kw">length</span>(mt_idx)<span class="op">!=</span><span class="dv">0</span>){</a>
<a class="sourceLine" id="cb52-9" title="9">    sp_count    &lt;-<span class="st"> </span>sp_count[<span class="op">-</span>mt_idx,]</a>
<a class="sourceLine" id="cb52-10" title="10">}</a>
<a class="sourceLine" id="cb52-11" title="11">sparkX &lt;-<span class="st"> </span><span class="kw">sparkx</span>(sp_count,location,<span class="dt">numCores=</span><span class="dv">1</span>,<span class="dt">option=</span><span class="st">&quot;mixture&quot;</span>)</a>
<a class="sourceLine" id="cb52-12" title="12"><span class="kw">head</span>(sparkX<span class="op">$</span>res_mtest)</a>
<a class="sourceLine" id="cb52-13" title="13">qvals[,<span class="st">&quot;SPARK-X&quot;</span>] &lt;-sparkX<span class="op">$</span>res_mtest<span class="op">$</span>adjustedPval</a></code></pre></div>
<p>Since we manually created non-SVG in the <code>extra_para()</code> step, now we can calculate the actual false discovery proportion(FDP) and power of the tests we conducted above with various target FDR threshold.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" title="1">test =<span class="st"> </span><span class="kw">colnames</span>(qvals)</a>
<a class="sourceLine" id="cb53-2" title="2">targetFDR &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="fl">0.01</span>,<span class="fl">0.05</span>,<span class="fl">0.1</span>,<span class="fl">0.2</span>,<span class="fl">0.5</span>)</a>
<a class="sourceLine" id="cb53-3" title="3">de &lt;-<span class="st"> </span>de_idx</a>
<a class="sourceLine" id="cb53-4" title="4">fdp_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(targetFDR), <span class="dt">ncol =</span> <span class="kw">length</span>(test))</a>
<a class="sourceLine" id="cb53-5" title="5"><span class="kw">colnames</span>(fdp_mat) &lt;-<span class="st"> </span>test</a>
<a class="sourceLine" id="cb53-6" title="6"><span class="kw">rownames</span>(fdp_mat) &lt;-<span class="st"> </span>targetFDR</a>
<a class="sourceLine" id="cb53-7" title="7">power_mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dv">0</span>, <span class="dt">nrow =</span> <span class="kw">length</span>(targetFDR), <span class="dt">ncol =</span> <span class="kw">length</span>(test))</a>
<a class="sourceLine" id="cb53-8" title="8"><span class="kw">colnames</span>(power_mat) &lt;-<span class="st"> </span>test</a>
<a class="sourceLine" id="cb53-9" title="9"><span class="kw">rownames</span>(power_mat) &lt;-<span class="st"> </span>targetFDR</a>
<a class="sourceLine" id="cb53-10" title="10"></a>
<a class="sourceLine" id="cb53-11" title="11"><span class="cf">for</span> (t <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(test)) {</a>
<a class="sourceLine" id="cb53-12" title="12">  curr_p &lt;-<span class="st"> </span>qvals[,t]</a>
<a class="sourceLine" id="cb53-13" title="13">  curr_test &lt;-<span class="st"> </span>test[t]</a>
<a class="sourceLine" id="cb53-14" title="14">  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(targetFDR)) {</a>
<a class="sourceLine" id="cb53-15" title="15">    thre &lt;-<span class="st"> </span>targetFDR[i]</a>
<a class="sourceLine" id="cb53-16" title="16">    <span class="cf">if</span>(curr_test <span class="op">==</span><span class="st"> &quot;spatialDE&quot;</span>){</a>
<a class="sourceLine" id="cb53-17" title="17">      de_results &lt;-<span class="st"> </span>results_spatialDE[results_spatialDE<span class="op">$</span>qval <span class="op">&lt;</span><span class="st"> </span>thre, ]</a>
<a class="sourceLine" id="cb53-18" title="18">      ms_results &lt;-<span class="st"> </span><span class="kw">model_search</span>(</a>
<a class="sourceLine" id="cb53-19" title="19">      resid_expr,</a>
<a class="sourceLine" id="cb53-20" title="20">      <span class="dt">coordinates =</span> X,</a>
<a class="sourceLine" id="cb53-21" title="21">      <span class="dt">de_results =</span> de_results</a>
<a class="sourceLine" id="cb53-22" title="22">      )</a>
<a class="sourceLine" id="cb53-23" title="23">      discovery &lt;-<span class="st"> </span>ms_results<span class="op">$</span>g</a>
<a class="sourceLine" id="cb53-24" title="24">      tp &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(ms_results<span class="op">$</span>g,de))</a>
<a class="sourceLine" id="cb53-25" title="25">    }<span class="cf">else</span>{</a>
<a class="sourceLine" id="cb53-26" title="26">      discovery &lt;-<span class="st"> </span><span class="kw">which</span>(curr_p <span class="op">&lt;=</span><span class="st"> </span>thre)</a>
<a class="sourceLine" id="cb53-27" title="27">      tp &lt;-<span class="st"> </span><span class="kw">length</span>(<span class="kw">intersect</span>(<span class="kw">names</span>(discovery),de))</a>
<a class="sourceLine" id="cb53-28" title="28">    }</a>
<a class="sourceLine" id="cb53-29" title="29">    <span class="cf">if</span>(<span class="kw">length</span>(discovery) <span class="op">==</span><span class="st"> </span><span class="dv">0</span>){</a>
<a class="sourceLine" id="cb53-30" title="30">      fdp &lt;-<span class="st"> </span><span class="dv">0</span></a>
<a class="sourceLine" id="cb53-31" title="31">    }<span class="cf">else</span>{</a>
<a class="sourceLine" id="cb53-32" title="32">      fdp &lt;-<span class="st"> </span>(<span class="kw">length</span>(discovery) <span class="op">-</span><span class="st"> </span>tp)<span class="op">/</span><span class="kw">length</span>(discovery)</a>
<a class="sourceLine" id="cb53-33" title="33">    }</a>
<a class="sourceLine" id="cb53-34" title="34">    power &lt;-<span class="st"> </span>tp<span class="op">/</span><span class="kw">length</span>(de)</a>
<a class="sourceLine" id="cb53-35" title="35">    fdp_mat[i, t] &lt;-<span class="st"> </span>fdp</a>
<a class="sourceLine" id="cb53-36" title="36">    power_mat[i,t] &lt;-<span class="st"> </span>power</a>
<a class="sourceLine" id="cb53-37" title="37">  }</a>
<a class="sourceLine" id="cb53-38" title="38">}</a></code></pre></div>
<p>Lastly, we visualize the Target FDR vs Actual FDP and Target FDR vs Power below.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" title="1">fdp_long &lt;-<span class="st"> </span><span class="kw">melt</span>(fdp_mat)</a>
<a class="sourceLine" id="cb54-2" title="2"><span class="kw">colnames</span>(fdp_long) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Target FDR&quot;</span>,<span class="st">&quot;test_method&quot;</span>,<span class="st">&quot;Actual FDP&quot;</span>)</a>
<a class="sourceLine" id="cb54-3" title="3">fdp_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(fdp_long) <span class="op">+</span></a>
<a class="sourceLine" id="cb54-4" title="4"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span><span class="st">`</span><span class="dt">Actual FDP</span><span class="st">`</span>,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb54-5" title="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span><span class="st">`</span><span class="dt">Actual FDP</span><span class="st">`</span>,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb54-6" title="6"><span class="st">  </span><span class="kw">geom_abline</span>(<span class="dt">intercept =</span> <span class="dv">0</span>, <span class="dt">slope=</span><span class="dv">1</span>,<span class="dt">linetype=</span><span class="st">&quot;dashed&quot;</span>,<span class="dt">color=</span><span class="st">&quot;grey&quot;</span>)<span class="op">+</span><span class="st"> </span></a>
<a class="sourceLine" id="cb54-7" title="7"><span class="st"> </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>) <span class="op">+</span><span class="st"> </span><span class="kw">expand_limits</span>(<span class="dt">x =</span> <span class="dv">0</span>, <span class="dt">y =</span> <span class="kw">c</span>(<span class="dv">0</span>,<span class="dv">1</span>))</a>
<a class="sourceLine" id="cb54-8" title="8">fdp_plot</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-56-1.png" width="700" /></p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" title="1">power_long &lt;-<span class="st"> </span><span class="kw">melt</span>(power_mat)</a>
<a class="sourceLine" id="cb55-2" title="2"><span class="kw">colnames</span>(power_long) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;Target FDR&quot;</span>,<span class="st">&quot;test_method&quot;</span>,<span class="st">&quot;Power&quot;</span>)</a>
<a class="sourceLine" id="cb55-3" title="3">power_plot &lt;-<span class="st"> </span><span class="kw">ggplot</span>(power_long) <span class="op">+</span></a>
<a class="sourceLine" id="cb55-4" title="4"><span class="st">  </span><span class="kw">geom_line</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span>Power,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb55-5" title="5"><span class="st">  </span><span class="kw">geom_point</span>(<span class="kw">aes</span>(<span class="dt">x=</span><span class="st">`</span><span class="dt">Target FDR</span><span class="st">`</span>, <span class="dt">y=</span>Power,<span class="dt">color=</span>test_method))<span class="op">+</span></a>
<a class="sourceLine" id="cb55-6" title="6"><span class="st">  </span><span class="kw">theme</span>(<span class="dt">aspect.ratio =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb55-7" title="7">power_plot</a></code></pre></div>
<p><img src="scDesign3-DEanalysis-vignette_files/figure-html/unnamed-chunk-57-1.png" width="700" /></p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>



    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Dongyuan Song, Qingyang Wang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>
