<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="scDesign3">
<title>Simulate datasets with cell type and modified cell-type proportions  â€¢ scDesign3</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Simulate datasets with cell type and modified cell-type proportions ">
<meta property="og:description" content="scDesign3">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">scDesign3</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/scDesign3.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/scDesign3-introduction-vignette.html">scDesign3 Introduction</a>
    <a class="dropdown-item" href="../articles/scDesign3-marginal-vignette.html">scDesign3 marginal distribution for genes</a>
    <a class="dropdown-item" href="../articles/scDesign3-copulaCompare-vignette.html">Compare gaussian copula and vine copula</a>
    <a class="dropdown-item" href="../articles/scDesign3-parallelization-vignette.html">Parallelization in scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-cellType-vignette.html">Simulate datasets with cell type and modified cell-type proportions </a>
    <a class="dropdown-item" href="../articles/scDesign3-librarySize-vignette.html">Simulate datasets with cell library size</a>
    <a class="dropdown-item" href="../articles/scDesign3-multipleLineages-vignette.html">Simulate datasets with multiple lineages</a>
    <a class="dropdown-item" href="../articles/scDesign3-conditionEffect-vignette.html">Simulate datasets with condition effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-batchEffect-vignette.html">Simulate datasets with batch effect</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-vignette.html">Simulate spatial transcriptomic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-spatial-deconvolution.html">Simulate spot-resolution spatial data for cell-type deconvolution</a>
    <a class="dropdown-item" href="../articles/scDesign3-scATACseq-vignette.html">Simulate single-cell ATAC-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-CITEseq-vignette.html">Simulate CITE-seq data</a>
    <a class="dropdown-item" href="../articles/scDesign3-multiomics-vignette.html">Simulate multi-omics data from single-omic data</a>
    <a class="dropdown-item" href="../articles/scDesign3-clusterGOF-vignette.html">Evaluate clustering goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-pseudotimeGOF-vignette.html">Evaluate pseudotime goodness-of-fit by scDesign3</a>
    <a class="dropdown-item" href="../articles/scDesign3-DEanalysis-vignette.html">Benchmarking DE Analysis with scDesign3</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/SONGDONGYUAN1994/scDesign3/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Simulate datasets with cell type and modified cell-type proportions </h1>
                        <h4 data-toc-skip class="author">Dongyuan
Song</h4>
            <address class="author_afil">
      Bioinformatics IDP, University of California, Los
Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:dongyuansong@ucla.edu" class="email">dongyuansong@ucla.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Qingyang
Wang</h4>
            <address class="author_afil">
      Department of Statistics, University of California, Los
Angeles<br><a class="author_email" href="mailto:#"></a><a href="mailto:qw802@g.ucla.edu" class="email">qw802@g.ucla.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">21 August 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/SONGDONGYUAN1994/scDesign3/blob/HEAD/../../vignettes/scDesign3-cellType-vignette.Rmd" class="external-link"><code>../../vignettes/scDesign3-cellType-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>scDesign3-cellType-vignette.Rmd</code></div>
    </div>

    
    
<style type="text/css">
pre {
  white-space: pre !important;
  overflow-x: scroll !important;
}
</style>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/SONGDONGYUAN1994/scDesign3" class="external-link">scDesign3</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">DuoClustering2018</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">scran</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html" class="external-link">theme_set</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this tutorial, we will demonstrate how to use scDesign3 to
simulate datasets with cell-type labels as covariates. We will also
guide you through modifying cell-type proportions and generating
corresponding synthetic data using our step-by-step functions.</p>
</div>
<div class="section level2">
<h2 id="read-in-the-reference-data">Read in the reference data<a class="anchor" aria-label="anchor" href="#read-in-the-reference-data"></a>
</h2>
<p>The raw data is from the R package <code>DuoClustering2018</code>
which contain a set of datasets with true cell type labels.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/get.html" class="external-link">get</a></span><span class="op">(</span><span class="st">"sce_filteredExpr10_Zhengmix4eq"</span><span class="op">)</span><span class="op">(</span>metadata <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">cell_type</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">as.factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">phenoid</span><span class="op">)</span></span></code></pre></div>
<p>To save computational time, we only use the top 200 highly variable
genes.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ngene</span> <span class="op">&lt;-</span> <span class="fl">200</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">temp_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/modelGeneVar.html" class="external-link">modelGeneVar</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span>
<span><span class="va">chosen</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scran/man/getTopHVGs.html" class="external-link">getTopHVGs</a></span><span class="op">(</span><span class="va">temp_sce</span>, n <span class="op">=</span> <span class="va">ngene</span><span class="op">)</span></span>
<span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="va">sce</span><span class="op">[</span><span class="va">chosen</span>,<span class="op">]</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulation-with-original-cell-type-labels">Simulation with original cell-type labels<a class="anchor" aria-label="anchor" href="#simulation-with-original-cell-type-labels"></a>
</h2>
<p>Then, we set the parameter <code>celltype = "cell_type"</code>,
indicating that the cell-type labels are stored in the
<code>cell_type</code> column within the <code>colData</code> of the
input <code>SingleCellExperiment</code> object. We also set the
parameters <code>mu_formula</code> and <code>sigma_formula</code> as
<code>cell_type</code>, specifying that in the GAMLSS model, both the
log-transformed mean and dispersion parameters for each geneâ€™s marginal
distribution (in this case, the negative binomial distribution) have a
linear relationship with the covariate <code>cell_type</code>. The
parameter <code>corr_formula</code> is set to <code>cell_type</code>,
meaning that each cell type will have its own unique gene-gene
correlation structure.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">example_simu</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/scdesign3.html">scdesign3</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    celltype <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    pseudotime <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    spatial <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    other_covariates <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    mu_formula <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    sigma_formula <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    usebam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    corr_formula <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    copula <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>    DT <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    pseudo_obs <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    return_model <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    nonzerovar <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    parallelization <span class="op">=</span> <span class="st">"pbmcmapply"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Then, we can create the <code>SinglecellExperiment</code> object
using the synthetic count matrix and store the <code>logcounts</code> to
the input and synthetic <code>SinglecellExperiment</code> objects.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">simu_sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">example_simu</span><span class="op">$</span><span class="va">new_count</span><span class="op">)</span>, colData <span class="op">=</span> <span class="va">example_simu</span><span class="op">$</span><span class="va">new_covariate</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">simu_sce</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="simulation-with-modified-cell-type-proportions">Simulation with modified cell-type proportions<a class="anchor" aria-label="anchor" href="#simulation-with-modified-cell-type-proportions"></a>
</h2>
<p>To change the cell-type proportions in the simulated dataset, we need
to use scDesign3â€™s step-by-step functions. Specifically, we provide a
new set of cell-type labels with the desired proportions in the
<code><a href="../reference/extract_para.html">extract_para()</a></code> and <code><a href="../reference/simu_new.html">simu_new()</a></code> functions. The
detailed code is shown below.</p>
<div class="section level3">
<h3 id="construct-the-input-dataset-">1. Construct the input dataset.<a class="anchor" aria-label="anchor" href="#construct-the-input-dataset-"></a>
</h3>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/construct_data.html">construct_data</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    celltype <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    pseudotime <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    spatial <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    other_covariates <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    corr_by <span class="op">=</span> <span class="st">"cell_type"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fit-regression-models-for-each-feature-based-on-your-specification-">2. Fit regression models for each feature based on your
specification.<a class="anchor" aria-label="anchor" href="#fit-regression-models-for-each-feature-based-on-your-specification-"></a>
</h3>
<p>Here, we set the parameters <code>mu_formula</code> and
<code>sigma_formula</code> as <code>cell_type</code>, indicating that in
the GAMLSS model, the log-transformed mean and dispersion parameters for
each geneâ€™s marginal distribution (in this case, the negative binomial
distribution) has a linear relationship with the covariate
<code>cell_type</code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_marginal</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_marginal.html">fit_marginal</a></span><span class="op">(</span></span>
<span>    data <span class="op">=</span> <span class="va">example_data</span>,</span>
<span>    predictor <span class="op">=</span> <span class="st">"gene"</span>,</span>
<span>    mu_formula <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    sigma_formula <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    usebam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    parallelization <span class="op">=</span> <span class="st">"pbmcmapply"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="fit-a-copula-obtain-aic-and-bic-">3. Fit a copula, obtain AIC and BIC.<a class="anchor" aria-label="anchor" href="#fit-a-copula-obtain-aic-and-bic-"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">example_copula</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/fit_copula.html">fit_copula</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    assay_use <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">example_marginal</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    copula <span class="op">=</span> <span class="st">"gaussian"</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">2</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="extract-out-the-estimated-parameters-so-you-can-make-some-modifications-and-use-the-modified-parameters-to-generate-new-data-if-needed-">4. Extract out the estimated parameters so you can make some
modifications and use the modified parameters to generate new data if
needed.<a class="anchor" aria-label="anchor" href="#extract-out-the-estimated-parameters-so-you-can-make-some-modifications-and-use-the-modified-parameters-to-generate-new-data-if-needed-"></a>
</h3>
<p>The cell-type proportions in the inputted dataset are shown below</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         b.cells  cd14.monocytes naive.cytotoxic    regulatory.t </span></span>
<span><span class="co">#&gt;       0.2511955       0.2447257       0.2483826       0.2556962</span></span></code></pre></div>
<p>For demonstration purpose, we generate a set of cell-type labels with
0% of b cells, 0% of CD 14 monocytes, 20% of naive cytotoxic cell, and
80% regulartory T cells. We will supply the data frame
<code>new_ct</code> generated below to the <code>new_covariate</code>
parameter in <code><a href="../reference/extract_para.html">extract_para()</a></code> and <code><a href="../reference/simu_new.html">simu_new()</a></code>.
The <code>new_covariate</code> parameters must include all the
covariates specified in <code>mu_formula</code> and
<code>sigma_formula</code>, and a column called <code>corr_group</code>
indicating the gene-gene correlation structure. Here, since we specify
<code>corr_formula = "cell_type"</code> in
<code><a href="../reference/construct_data.html">construct_data()</a></code>, the <code>corr_group</code> column in
<code>new_ct</code> also need to reflect this.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ct_prop</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0.2</span>,<span class="fl">0.8</span><span class="op">)</span></span>
<span><span class="va">unique_combined</span> <span class="op">&lt;-</span> <span class="va">example_data</span><span class="op">$</span><span class="va">dat</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/expand.html" class="external-link">expand</a></span><span class="op">(</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/expand.html" class="external-link">nesting</a></span><span class="op">(</span><span class="va">cell_type</span>, <span class="va">corr_group</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">new_ct</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">unique_combined</span>, <span class="va">rep</span>,<span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">ct_prop</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">new_ct</span><span class="op">)</span></span>
<span><span class="co">#&gt;         cell_type      corr_group</span></span>
<span><span class="co">#&gt; 1 naive.cytotoxic naive.cytotoxic</span></span>
<span><span class="co">#&gt; 2 naive.cytotoxic naive.cytotoxic</span></span>
<span><span class="co">#&gt; 3 naive.cytotoxic naive.cytotoxic</span></span>
<span><span class="co">#&gt; 4 naive.cytotoxic naive.cytotoxic</span></span>
<span><span class="co">#&gt; 5 naive.cytotoxic naive.cytotoxic</span></span>
<span><span class="co">#&gt; 6 naive.cytotoxic naive.cytotoxic</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">new_ct</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">new_ct</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;         b.cells  cd14.monocytes naive.cytotoxic    regulatory.t </span></span>
<span><span class="co">#&gt;             0.0             0.0             0.2             0.8</span></span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">example_para</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/extract_para.html">extract_para</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    marginal_list <span class="op">=</span> <span class="va">example_marginal</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="va">new_ct</span>,</span>
<span>    data <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">dat</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="simulate-the-new-count-matrix-">5. Simulate the new count matrix.<a class="anchor" aria-label="anchor" href="#simulate-the-new-count-matrix-"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">example_newcount</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simu_new.html">simu_new</a></span><span class="op">(</span></span>
<span>    sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>    mean_mat <span class="op">=</span> <span class="va">example_para</span><span class="op">$</span><span class="va">mean_mat</span>,</span>
<span>    sigma_mat <span class="op">=</span> <span class="va">example_para</span><span class="op">$</span><span class="va">sigma_mat</span>,</span>
<span>    zero_mat <span class="op">=</span> <span class="va">example_para</span><span class="op">$</span><span class="va">zero_mat</span>,</span>
<span>    quantile_mat <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>    copula_list <span class="op">=</span> <span class="va">example_copula</span><span class="op">$</span><span class="va">copula_list</span>,</span>
<span>    n_cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>    family_use <span class="op">=</span> <span class="st">"nb"</span>,</span>
<span>    input_data <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">dat</span>,</span>
<span>    new_covariate <span class="op">=</span> <span class="va">new_ct</span>,</span>
<span>    important_feature <span class="op">=</span> <span class="va">example_copula</span><span class="op">$</span><span class="va">important_feature</span>,</span>
<span>    filtered_gene <span class="op">=</span> <span class="va">example_data</span><span class="op">$</span><span class="va">filtered_gene</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>Then, we can create the <code>SinglecellExperiment</code> object
using the synthetic count matrix and store the <code>logcounts</code> to
the input and synthetic <code>SinglecellExperiment</code> objects.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">simu_sce2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/SingleCellExperiment.html" class="external-link">SingleCellExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">example_newcount</span><span class="op">)</span>, colData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>cell_type <span class="op">=</span> <span class="va">new_ct</span><span class="op">$</span><span class="va">cell_type</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/assays.html" class="external-link">logcounts</a></span><span class="op">(</span><span class="va">simu_sce2</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log1p</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">simu_sce2</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a>
</h2>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">compare_figure</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_reduceddim.html">plot_reduceddim</a></span><span class="op">(</span>ref_sce <span class="op">=</span> <span class="va">sce</span>, </span>
<span>                                  sce_list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">simu_sce</span>, <span class="va">simu_sce2</span><span class="op">)</span>, </span>
<span>                                  name_vec <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Reference"</span>, <span class="st">"Same cell-type proportions"</span>, <span class="st">"Modified cell-type proportions"</span><span class="op">)</span>,</span>
<span>                                  assay_use <span class="op">=</span> <span class="st">"logcounts"</span>, </span>
<span>                                  if_plot <span class="op">=</span> <span class="cn">TRUE</span>, </span>
<span>                                  color_by <span class="op">=</span> <span class="st">"cell_type"</span>, </span>
<span>                                  n_pc <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">compare_figure</span><span class="op">$</span><span class="va">p_umap</span><span class="op">)</span></span></code></pre></div>
<p><img src="scDesign3-cellType-vignette_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3;  LAPACK version 3.9.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: America/Los_Angeles</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] lubridate_1.9.2             forcats_1.0.0              </span></span>
<span><span class="co">#&gt;  [3] stringr_1.5.1               dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [5] purrr_1.0.1                 readr_2.1.4                </span></span>
<span><span class="co">#&gt;  [7] tidyr_1.3.0                 tibble_3.2.1               </span></span>
<span><span class="co">#&gt;  [9] tidyverse_2.0.0             scran_1.28.2               </span></span>
<span><span class="co">#&gt; [11] scuttle_1.10.1              DuoClustering2018_1.18.0   </span></span>
<span><span class="co">#&gt; [13] ggplot2_3.5.0               SingleCellExperiment_1.22.0</span></span>
<span><span class="co">#&gt; [15] SummarizedExperiment_1.30.2 Biobase_2.60.0             </span></span>
<span><span class="co">#&gt; [17] GenomicRanges_1.52.0        GenomeInfoDb_1.36.1        </span></span>
<span><span class="co">#&gt; [19] IRanges_2.34.1              S4Vectors_0.38.1           </span></span>
<span><span class="co">#&gt; [21] BiocGenerics_0.46.0         MatrixGenerics_1.12.2      </span></span>
<span><span class="co">#&gt; [23] matrixStats_1.0.0           scDesign3_1.1.3            </span></span>
<span><span class="co">#&gt; [25] BiocStyle_2.28.0           </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.4.0                 later_1.3.1                  </span></span>
<span><span class="co">#&gt;   [3] bitops_1.0-7                  filelock_1.0.2               </span></span>
<span><span class="co">#&gt;   [5] gamlss.data_6.0-2             lifecycle_1.0.4              </span></span>
<span><span class="co">#&gt;   [7] pbmcapply_1.5.1               edgeR_3.42.4                 </span></span>
<span><span class="co">#&gt;   [9] rprojroot_2.0.3               lattice_0.22-5               </span></span>
<span><span class="co">#&gt;  [11] MASS_7.3-60.0.1               magrittr_2.0.3               </span></span>
<span><span class="co">#&gt;  [13] limma_3.56.2                  sass_0.4.7                   </span></span>
<span><span class="co">#&gt;  [15] rmarkdown_2.23                jquerylib_0.1.4              </span></span>
<span><span class="co">#&gt;  [17] yaml_2.3.7                    metapod_1.8.0                </span></span>
<span><span class="co">#&gt;  [19] httpuv_1.6.11                 askpass_1.1                  </span></span>
<span><span class="co">#&gt;  [21] reticulate_1.30               DBI_1.1.3                    </span></span>
<span><span class="co">#&gt;  [23] abind_1.4-5                   zlibbioc_1.46.0              </span></span>
<span><span class="co">#&gt;  [25] RCurl_1.98-1.12               rappdirs_0.3.3               </span></span>
<span><span class="co">#&gt;  [27] GenomeInfoDbData_1.2.10       irlba_2.3.5.1                </span></span>
<span><span class="co">#&gt;  [29] umap_0.2.10.0                 RSpectra_0.16-2              </span></span>
<span><span class="co">#&gt;  [31] dqrng_0.3.0                   pkgdown_2.0.7                </span></span>
<span><span class="co">#&gt;  [33] DelayedMatrixStats_1.22.1     codetools_0.2-19             </span></span>
<span><span class="co">#&gt;  [35] DelayedArray_0.26.6           tidyselect_1.2.0             </span></span>
<span><span class="co">#&gt;  [37] farver_2.1.1                  ScaledMatrix_1.8.1           </span></span>
<span><span class="co">#&gt;  [39] viridis_0.6.4                 BiocFileCache_2.8.0          </span></span>
<span><span class="co">#&gt;  [41] jsonlite_1.8.7                BiocNeighbors_1.18.0         </span></span>
<span><span class="co">#&gt;  [43] ellipsis_0.3.2                survival_3.5-8               </span></span>
<span><span class="co">#&gt;  [45] systemfonts_1.0.4             tools_4.4.0                  </span></span>
<span><span class="co">#&gt;  [47] ragg_1.2.5                    Rcpp_1.0.12                  </span></span>
<span><span class="co">#&gt;  [49] glue_1.7.0                    gridExtra_2.3                </span></span>
<span><span class="co">#&gt;  [51] xfun_0.39                     mgcv_1.9-1                   </span></span>
<span><span class="co">#&gt;  [53] ggthemes_4.2.4                withr_3.0.0                  </span></span>
<span><span class="co">#&gt;  [55] BiocManager_1.30.21.1         fastmap_1.1.1                </span></span>
<span><span class="co">#&gt;  [57] bluster_1.10.0                fansi_1.0.6                  </span></span>
<span><span class="co">#&gt;  [59] openssl_2.1.0                 digest_0.6.33                </span></span>
<span><span class="co">#&gt;  [61] rsvd_1.0.5                    gamlss_5.4-12                </span></span>
<span><span class="co">#&gt;  [63] timechange_0.2.0              R6_2.5.1                     </span></span>
<span><span class="co">#&gt;  [65] mime_0.12                     textshaping_0.3.6            </span></span>
<span><span class="co">#&gt;  [67] colorspace_2.1-0              RSQLite_2.3.1                </span></span>
<span><span class="co">#&gt;  [69] utf8_1.2.4                    generics_0.1.3               </span></span>
<span><span class="co">#&gt;  [71] httr_1.4.6                    S4Arrays_1.2.0               </span></span>
<span><span class="co">#&gt;  [73] pkgconfig_2.0.3               gtable_0.3.4                 </span></span>
<span><span class="co">#&gt;  [75] blob_1.2.4                    XVector_0.40.0               </span></span>
<span><span class="co">#&gt;  [77] htmltools_0.5.5               bookdown_0.34                </span></span>
<span><span class="co">#&gt;  [79] scales_1.3.0                  png_0.1-8                    </span></span>
<span><span class="co">#&gt;  [81] knitr_1.43                    rstudioapi_0.15.0            </span></span>
<span><span class="co">#&gt;  [83] tzdb_0.4.0                    reshape2_1.4.4               </span></span>
<span><span class="co">#&gt;  [85] nlme_3.1-163                  curl_5.0.1                   </span></span>
<span><span class="co">#&gt;  [87] cachem_1.0.8                  BiocVersion_3.17.1           </span></span>
<span><span class="co">#&gt;  [89] parallel_4.4.0                AnnotationDbi_1.62.2         </span></span>
<span><span class="co">#&gt;  [91] desc_1.4.2                    pillar_1.9.0                 </span></span>
<span><span class="co">#&gt;  [93] grid_4.4.0                    vctrs_0.6.5                  </span></span>
<span><span class="co">#&gt;  [95] promises_1.2.0.1              BiocSingular_1.16.0          </span></span>
<span><span class="co">#&gt;  [97] dbplyr_2.3.3                  beachmat_2.16.0              </span></span>
<span><span class="co">#&gt;  [99] xtable_1.8-4                  cluster_2.1.6                </span></span>
<span><span class="co">#&gt; [101] gamlss.dist_6.0-5             evaluate_0.21                </span></span>
<span><span class="co">#&gt; [103] mvtnorm_1.2-2                 cli_3.6.2                    </span></span>
<span><span class="co">#&gt; [105] locfit_1.5-9.8                compiler_4.4.0               </span></span>
<span><span class="co">#&gt; [107] rlang_1.1.3                   crayon_1.5.2                 </span></span>
<span><span class="co">#&gt; [109] labeling_0.4.3                mclust_6.0.1                 </span></span>
<span><span class="co">#&gt; [111] plyr_1.8.9                    fs_1.6.3                     </span></span>
<span><span class="co">#&gt; [113] stringi_1.8.3                 viridisLite_0.4.2            </span></span>
<span><span class="co">#&gt; [115] BiocParallel_1.34.2           munsell_0.5.0                </span></span>
<span><span class="co">#&gt; [117] Biostrings_2.68.1             coop_0.6-3                   </span></span>
<span><span class="co">#&gt; [119] Matrix_1.7-0                  ExperimentHub_2.8.1          </span></span>
<span><span class="co">#&gt; [121] hms_1.1.3                     sparseMatrixStats_1.12.2     </span></span>
<span><span class="co">#&gt; [123] bit64_4.0.5                   KEGGREST_1.40.0              </span></span>
<span><span class="co">#&gt; [125] statmod_1.5.0                 shiny_1.7.4.1                </span></span>
<span><span class="co">#&gt; [127] interactiveDisplayBase_1.38.0 highr_0.10                   </span></span>
<span><span class="co">#&gt; [129] AnnotationHub_3.8.0           igraph_1.5.0.1               </span></span>
<span><span class="co">#&gt; [131] memoise_2.0.1                 bslib_0.5.0                  </span></span>
<span><span class="co">#&gt; [133] bit_4.0.5</span></span></code></pre></div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Dongyuan Song, Qingyang Wang, Chenxin Jiang.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
